diff -Nur qstardict-0.13.1.orig/plugins/stardict/dictziplib.cpp qstardict-0.13.1/plugins/stardict/dictziplib.cpp
--- qstardict-0.13.1.orig/plugins/stardict/dictziplib.cpp	2009-02-10 13:33:29.000000000 +0100
+++ qstardict-0.13.1/plugins/stardict/dictziplib.cpp	2009-06-02 21:13:50.316497364 +0200
@@ -305,7 +305,7 @@
 
 #ifdef Q_OS_UNIX
     if (stat(fname.c_str(), &sb) || !S_ISREG(sb.st_mode))
-#elif def Q_OS_WIN32
+#elif defined(Q_OS_WIN32)
     if (_stat(fname.c_str(), &sb) || !(sb.stMode & _S_IFREG))
 #endif
     {
