diff -Nru base0123/src/emu/inptport.c w0123/src/emu/inptport.c
--- base0123/src/emu/inptport.c	2008-01-30 12:27:24.000000000 +1300
+++ w0123/src/emu/inptport.c	2008-02-06 01:24:34.000000000 +1300
@@ -90,6 +90,7 @@
     except that the accumulated deltas are not bounded, but rather wrap.
 
 ***************************************************************************/
+#define _USE_32BIT_TIME_T  // VC2005 uses a 64-bit time_t structure by default
 
 #include "osdepend.h"
 #include "driver.h"
@@ -261,10 +262,10 @@
 static input_port_entry *input_ports_default;
 
 /* recorded speed read from an INP file */
-static double rec_speed;
+double rec_speed;
 
 /* set to TRUE if INP file being played is an extended INP file */
-static int extended_inp;
+int extended_inp;
 
 /* for average speed calculations */
 static int framecount;
@@ -314,8 +315,6 @@
 	input_port_28_dword_r,	input_port_29_dword_r,	input_port_30_dword_r,	input_port_31_dword_r
 };
 
-
-
 /***************************************************************************
     COMMON SHARED STRINGS
 ***************************************************************************/
@@ -437,7 +436,6 @@
 };
 
 
-
 /***************************************************************************
     DEFAULT INPUT PORTS
 ***************************************************************************/
@@ -1167,7 +1165,7 @@
 
 	/* Check if input file is an eXtended INP file */
 	extended_inp = (memcmp(check, "XINP\0\0\0", 7) == 0);
-	if (extended_inp)
+	if (!extended_inp)
 	{
 		/* read playback header */
 		mame_fread(machine->playback_file, &inpheader, sizeof(inpheader));
@@ -1213,8 +1211,9 @@
 static void setup_record(running_machine *machine)
 {
 	const char *filename = options_get_string(mame_options(), OPTION_RECORD);
-	inp_header inpheader;
+	//inp_header inpheader;
 	file_error filerr;
+	ext_inp_header xheader;
 
 	/* if no file, nothing to do */
 	if (filename[0] == 0)
@@ -1224,10 +1223,16 @@
 	filerr = mame_fopen(SEARCHPATH_INPUTLOG, filename, OPEN_FLAG_WRITE | OPEN_FLAG_CREATE | OPEN_FLAG_CREATE_PATHS, &machine->record_file);
 	assert_always(filerr == FILERR_NONE, "Failed to open file for recording");
 
+	/* disable cheats */
+	options_set_bool(mame_options(),OPTION_CHEAT,0,OPTION_PRIORITY_MAXIMUM);
+
 	/* create a header */
-	memset(&inpheader, 0, sizeof(inpheader));
-	strcpy(inpheader.name, machine->gamedrv->name);
-	mame_fwrite(machine->record_file, &inpheader, sizeof(inpheader));
+	memset(&xheader, '\0', sizeof(ext_inp_header));
+	strcpy(xheader.header, "XINP\0\0\0");
+	strcpy(xheader.shortname, machine->gamedrv->name);
+	strcpy(xheader.version, build_version);
+	xheader.starttime = (UINT32)time(NULL);
+	mame_fwrite(Machine->record_file, &xheader, sizeof(ext_inp_header));
 }
 
 
@@ -1243,8 +1248,13 @@
 	/* close any playback or recording files */
 	if (machine->playback_file != NULL)
 		mame_fclose(machine->playback_file);
-	if (machine->record_file != NULL)
-		mame_fclose(machine->record_file);
+  	if (machine->record_file != NULL)
+	{
+		UINT32 val = (UINT32)time(NULL);
+		mame_fseek(machine->record_file,52,SEEK_SET);
+		mame_fwrite(machine->record_file,&val,sizeof(UINT32));
+  		mame_fclose(machine->record_file);
+	}
 }
 
 
@@ -2786,7 +2796,7 @@
 			}
 		}
 	}
-
+	else  // because re-recording is cheating
 	/* handle recording */
 	if (Machine->record_file != NULL)
 	{
@@ -2959,7 +2969,7 @@
 				/* note that analog ports are handled instantaneously at port read time */
 			}
 	}
-
+	
 #ifdef MESS
 	/* less MESS to MESSy things */
 	inputx_update();
@@ -2994,6 +3004,15 @@
 			update_playback_record(portnum, readinputport(portnum));
 	}
 
+	/* record speed */
+	if(Machine->record_file != NULL)
+	{
+		double speed = video_get_speed_percent();
+		UINT32 marker = 0x00ABCDEF;  // end of frame marker
+		mame_fwrite(Machine->record_file,&speed,sizeof(double));
+		mame_fwrite(Machine->record_file,&marker,sizeof(UINT32));
+	}
+
 	/* store speed read from INP file, if extended INP */
 	if (Machine->playback_file != NULL && extended_inp)
 	{
diff -Nru base0123/src/emu/ui.c w0123/src/emu/ui.c
--- base0123/src/emu/ui.c	2008-02-02 21:57:39.000000000 +1300
+++ w0123/src/emu/ui.c	2008-02-06 01:02:33.000000000 +1300
@@ -317,7 +317,7 @@
 	int state;
 
 	/* disable everything if we are using -str */
-	if (!first_time || (str > 0 && str < 60*5) || Machine->gamedrv == &driver_empty)
+	if (!first_time || (str > 0 && str < 60*5) || Machine->gamedrv == &driver_empty  || Machine->record_file)
 		show_gameinfo = show_warnings = show_disclaimer = FALSE;
 
 	/* initialize the on-screen display system */
@@ -1256,7 +1256,7 @@
 		mame_schedule_soft_reset(Machine);
 
 	/* handle a request to display graphics/palette */
-	if (input_ui_pressed(IPT_UI_SHOW_GFX))
+	if (input_ui_pressed(IPT_UI_SHOW_GFX) && !Machine->record_file)
 	{
 		if (!is_paused)
 			mame_pause(Machine, TRUE);
@@ -1264,14 +1264,14 @@
 	}
 
 	/* handle a save state request */
-	if (input_ui_pressed(IPT_UI_SAVE_STATE))
+	if (input_ui_pressed(IPT_UI_SAVE_STATE) && !Machine->record_file)
 	{
 		mame_pause(Machine, TRUE);
 		return ui_set_handler(handler_load_save, LOADSAVE_SAVE);
 	}
 
 	/* handle a load state request */
-	if (input_ui_pressed(IPT_UI_LOAD_STATE))
+	if (input_ui_pressed(IPT_UI_LOAD_STATE) && !Machine->record_file)
 	{
 		mame_pause(Machine, TRUE);
 		return ui_set_handler(handler_load_save, LOADSAVE_LOAD);
@@ -1282,7 +1282,7 @@
 		video_save_active_screen_snapshots(Machine);
 
 	/* toggle pause */
-	if (input_ui_pressed(IPT_UI_PAUSE))
+	if (input_ui_pressed(IPT_UI_PAUSE) && !Machine->record_file)
 	{
 		/* with a shift key, it is single step */
 		if (is_paused && (input_code_pressed(KEYCODE_LSHIFT) || input_code_pressed(KEYCODE_RSHIFT)))
@@ -1348,11 +1348,11 @@
 	}
 
 	/* toggle throttle? */
-	if (input_ui_pressed(IPT_UI_THROTTLE))
+	if (input_ui_pressed(IPT_UI_THROTTLE) && !Machine->record_file)
 		video_set_throttle(!video_get_throttle());
 
 	/* check for fast forward */
-	if (input_port_type_pressed(IPT_UI_FAST_FORWARD, 0))
+	if (input_port_type_pressed(IPT_UI_FAST_FORWARD, 0) && !Machine->record_file)
 	{
 		video_set_fastforward(TRUE);
 		ui_show_fps_temp(0.5);
diff -Nru base0123/src/emu/uimenu.c w0123/src/emu/uimenu.c
--- base0123/src/emu/uimenu.c	2008-01-25 22:30:43.000000000 +1300
+++ w0123/src/emu/uimenu.c	2008-02-06 01:02:33.000000000 +1300
@@ -634,7 +634,7 @@
 		*selected = num_items - 1;
 
 	/* pause enables/disables pause */
-	if (input_ui_pressed(IPT_UI_PAUSE))
+	if (input_ui_pressed(IPT_UI_PAUSE) && !Machine->record_file)
 		mame_pause(Machine, !mame_is_paused(Machine));
 
 	return 0;
diff -Nru base0123/src/emu/video.c w0123/src/emu/video.c
--- base0123/src/emu/video.c	2008-02-03 18:23:32.000000000 +1300
+++ w0123/src/emu/video.c	2008-02-06 01:16:55.000000000 +1300
@@ -161,7 +161,9 @@
 	{ 0,1,1,1,1,1,1,1,1,1,1,1 }
 };
 
-
+// speed recorded in INP file
+extern double rec_speed;
+extern int extended_inp;
 
 /***************************************************************************
     FUNCTION PROTOTYPES
@@ -1267,6 +1269,10 @@
 	if (global.partial_updates_this_frame > 1)
 		dest += sprintf(dest, "\n%d partial updates", global.partial_updates_this_frame);
 
+	/* display recorded speed on playback */
+	if(Machine->playback_file != NULL && extended_inp)
+		dest += sprintf(dest,"\n Recorded speed %f%%", rec_speed);
+
 	/* return a pointer to the static buffer */
 	return buffer;
 }
@@ -2242,6 +2248,10 @@
 	}
 }
 
+double video_get_speed_percent(void)
+{
+	return global.speed_percent;
+}
 
 
 /***************************************************************************
diff -Nru base0123/src/emu/video.h w0123/src/emu/video.h
--- base0123/src/emu/video.h	2008-01-24 22:12:03.000000000 +1300
+++ w0123/src/emu/video.h	2008-02-06 01:02:33.000000000 +1300
@@ -166,4 +166,6 @@
 
 void video_crosshair_toggle(void);
 
+double video_get_speed_percent(void);
+
 #endif	/* __VIDEO_H__ */
