diff --git a/bsnes/Makefile b/bsnes/Makefile
index ec15fea..4331552 100755
--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -12,7 +12,7 @@ ui := ui
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects := libco
 
@@ -31,7 +31,7 @@ endif
 ifeq ($(platform),x)
   # tree vectorization causes code generation errors with Linux/GCC 4.6.1
   flags += -fno-tree-vectorize
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += $(if $(findstring console,$(options)),-mconsole,-mwindows)
