diff --git a/bsnes/target-libsnes/Makefile b/bsnes/target-libsnes/Makefile
index 378bcff..935d4bb 100755
--- a/bsnes/target-libsnes/Makefile
+++ b/bsnes/target-libsnes/Makefile
@@ -19,12 +19,12 @@ obj/libsnes.o: $(ui)/libsnes.cpp $(ui)/*
 build: $(objects)
 ifeq ($(platform),x)
 	ar rcs out/libsnes.a $(objects)
-	$(cpp) -o out/libsnes.so -shared -Wl,-soname,libsnes.so.1 $(objects)
+	$(cpp) -o out/libsnes.so -shared -Wl,-soname,libsnes.so.1 -Wl,--version-script=$(ui)/version-script $(objects)
 else ifeq ($(platform),osx)
 	ar rcs out/libsnes.a $(objects)
 	$(cpp) -o out/libsnes.dylib -install_name @executable_path/../Libraries/libsnes.dylib -shared -dynamiclib $(objects)
 else ifeq ($(platform),win)
-	$(cpp) -o out/snes.dll -shared -Wl,--out-implib,libsnes.a $(objects)
+	$(cpp) -o out/snes.dll -shared -Wl,--out-implib,libsnes.a -Wl,--version-script=$(ui)/version-script $(objects)
 endif
 
 install:
diff --git a/bsnes/target-libsnes/version-script b/bsnes/target-libsnes/version-script
new file mode 100644
index 0000000..9946df9
--- /dev/null
+++ b/bsnes/target-libsnes/version-script
@@ -0,0 +1,4 @@
+{
+    global: snes_*;
+    local: *;
+};
