diff --git a/bsnes/Makefile b/bsnes/Makefile
index eca3013..999a081 100755
--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -7,7 +7,7 @@ ui := ui
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects := libco
 
@@ -20,7 +20,7 @@ objects := libco
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows
