diff --git a/install.sh b/install.sh
index ae42a51..e5b4f65 100755
--- a/install.sh
+++ b/install.sh
@@ -12,17 +12,7 @@ MANPATH=${PREFIX}$3
 LIBPATH=${PREFIX}$4
 DOCPATH=${PREFIX}$5
 
-# this is not a perfect solution, but I don't know a better one at the moment:
-
-if [ -f /usr/X11R6/bin/mkdirhier ] ; then
-  MKDIRHIER="/usr/X11R6/bin/mkdirhier"
-else
-  if [ -f /usr/bin/X11/mkdirhier ] ; then
-    MKDIRHIER="/usr/bin/X11/mkdirhier"
-  else
-    MKDIRHIER="mkdir -p"
-  fi
-fi
+MKDIRHIER="mkdir -p"
 
 echo "Installing files:"
 
