diff --git a/kadu-core/CMakeLists.txt b/kadu-core/CMakeLists.txt
index 40ee01a..cc34c43 100644
--- a/kadu-core/CMakeLists.txt
+++ b/kadu-core/CMakeLists.txt
@@ -341,6 +341,7 @@ kadu_api_directories (kadu-core
 	os/qtsingleapplication
 	parser
 	plugin
+	plugin/activation
 	plugin/model
 	protocols
 	protocols/filter
@@ -348,6 +349,7 @@ kadu_api_directories (kadu-core
 	protocols/services
 	protocols/services/roster
 	provider
+	roster
 	qt
 	services
 	status
