diff --git a/kadu-core/CMakeLists.txt b/kadu-core/CMakeLists.txt
index 40ee01a..017769d 100644
--- a/kadu-core/CMakeLists.txt
+++ b/kadu-core/CMakeLists.txt
@@ -318,6 +318,7 @@ kadu_api_directories (kadu-core
 	gui/widgets/configuration
 	gui/widgets/dialog
 	gui/widgets/chat-widget
+	gui/widgets/group-tab-bar
 	gui/widgets/webkit-messages-view
 	chat-style/engine
 	gui/windows
@@ -341,6 +342,8 @@ kadu_api_directories (kadu-core
 	os/qtsingleapplication
 	parser
 	plugin
+	plugin/activation
+	plugin/metadata
 	plugin/model
 	protocols
 	protocols/filter
@@ -348,6 +351,7 @@ kadu_api_directories (kadu-core
 	protocols/services
 	protocols/services/roster
 	provider
+	roster
 	qt
 	services
 	status
