diff --git a/kadu-core/CMakeLists.txt b/kadu-core/CMakeLists.txt
index 0963518..1a5ca57 100644
--- a/kadu-core/CMakeLists.txt
+++ b/kadu-core/CMakeLists.txt
@@ -320,6 +320,9 @@ kadu_api_directories (kadu-core
 	gui/widgets
 	gui/widgets/configuration
 	gui/widgets/dialog
+	gui/widgets/chat-widget
+	gui/widgets/webkit-messages-view
+	chat-style/engine
 	gui/windows
 	gui/windows/open-chat-with
 	icons
