diff -Nur speedcrunch-0.9.orig/src/main/main.cpp speedcrunch-0.9/src/main/main.cpp
--- speedcrunch-0.9.orig/src/main/main.cpp	2007-11-16 01:31:28.000000000 +0100
+++ speedcrunch-0.9/src/main/main.cpp	2007-11-22 11:14:13.000000000 +0100
@@ -80,7 +80,7 @@
 
 
     // item (1)
-    QString shareDir = QString(br_find_data_dir(0)).append("/crunch");
+    QString shareDir = QString(br_find_data_dir(0)).append("/speedcrunch/translations");
  	QDir qmpath( shareDir );
  	qmpath.cd("crunch");
 
