diff --git a/modules/history/history_dialog.cpp b/modules/history/history_dialog.cpp
index daf5420..afa36e6 100644
--- a/modules/history/history_dialog.cpp
+++ b/modules/history/history_dialog.cpp
@@ -228,7 +228,7 @@ void HistoryDialog::rebuildIndex(bool)
 void HistoryDialog::showPopupMenu(const QPoint &pos)
 {
 	QTreeWidgetItem *item = uinsTreeWidget->itemAt(pos);
-	if (item->parent() == NULL) /* only for top level items */
+	if (item && item->parent() == NULL) /* only for top level items */
 		popupMenu->exec(uinsTreeWidget->mapToGlobal(pos));
 }
 
