diff --git a/plugins/history/storage/history-messages-storage.h b/plugins/history/storage/history-messages-storage.h
index 2d6a27c..e1d394e 100644
--- a/plugins/history/storage/history-messages-storage.h
+++ b/plugins/history/storage/history-messages-storage.h
@@ -27,8 +27,8 @@
 #include "message/message.h"
 #include "talkable/talkable.h"
 
-#include "plugins/history/history_exports.h"
-#include "plugins/history/history-query-result.h"
+#include "../history_exports.h"
+#include "../history-query-result.h"
 
 class HistoryQuery;
 
