diff -Nrub qmc2/minizip/ioapi.c qmc2-new/minizip/ioapi.c
--- qmc2/minizip/ioapi.c	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/ioapi.c	2011-11-13 22:20:10.000000000 -0500
@@ -29,38 +29,38 @@
 #define SEEK_SET    0
 #endif
 
-voidpf ZCALLBACK fopen_file_func OF((
+voidpf ZCALLBACK fopen_file_func _Z_OF((
    voidpf opaque,
    const char* filename,
    int mode));
 
-uLong ZCALLBACK fread_file_func OF((
+uLong ZCALLBACK fread_file_func _Z_OF((
    voidpf opaque,
    voidpf stream,
    void* buf,
    uLong size));
 
-uLong ZCALLBACK fwrite_file_func OF((
+uLong ZCALLBACK fwrite_file_func _Z_OF((
    voidpf opaque,
    voidpf stream,
    const void* buf,
    uLong size));
 
-long ZCALLBACK ftell_file_func OF((
+long ZCALLBACK ftell_file_func _Z_OF((
    voidpf opaque,
    voidpf stream));
 
-long ZCALLBACK fseek_file_func OF((
+long ZCALLBACK fseek_file_func _Z_OF((
    voidpf opaque,
    voidpf stream,
    uLong offset,
    int origin));
 
-int ZCALLBACK fclose_file_func OF((
+int ZCALLBACK fclose_file_func _Z_OF((
    voidpf opaque,
    voidpf stream));
 
-int ZCALLBACK ferror_file_func OF((
+int ZCALLBACK ferror_file_func _Z_OF((
    voidpf opaque,
    voidpf stream));
 
diff -Nrub qmc2/minizip/ioapi.h qmc2-new/minizip/ioapi.h
--- qmc2/minizip/ioapi.h	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/ioapi.h	2011-11-13 22:20:10.000000000 -0500
@@ -35,13 +35,13 @@
 extern "C" {
 #endif
 
-typedef voidpf (ZCALLBACK *open_file_func) OF((voidpf opaque, const char* filename, int mode));
-typedef uLong  (ZCALLBACK *read_file_func) OF((voidpf opaque, voidpf stream, void* buf, uLong size));
-typedef uLong  (ZCALLBACK *write_file_func) OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
-typedef long   (ZCALLBACK *tell_file_func) OF((voidpf opaque, voidpf stream));
-typedef long   (ZCALLBACK *seek_file_func) OF((voidpf opaque, voidpf stream, uLong offset, int origin));
-typedef int    (ZCALLBACK *close_file_func) OF((voidpf opaque, voidpf stream));
-typedef int    (ZCALLBACK *testerror_file_func) OF((voidpf opaque, voidpf stream));
+typedef voidpf (ZCALLBACK *open_file_func) _Z_OF((voidpf opaque, const char* filename, int mode));
+typedef uLong  (ZCALLBACK *read_file_func) _Z_OF((voidpf opaque, voidpf stream, void* buf, uLong size));
+typedef uLong  (ZCALLBACK *write_file_func) _Z_OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
+typedef long   (ZCALLBACK *tell_file_func) _Z_OF((voidpf opaque, voidpf stream));
+typedef long   (ZCALLBACK *seek_file_func) _Z_OF((voidpf opaque, voidpf stream, uLong offset, int origin));
+typedef int    (ZCALLBACK *close_file_func) _Z_OF((voidpf opaque, voidpf stream));
+typedef int    (ZCALLBACK *testerror_file_func) _Z_OF((voidpf opaque, voidpf stream));
 
 typedef struct zlib_filefunc_def_s
 {
@@ -57,7 +57,7 @@
 
 
 
-void fill_fopen_filefunc OF((zlib_filefunc_def* pzlib_filefunc_def));
+void fill_fopen_filefunc _Z_OF((zlib_filefunc_def* pzlib_filefunc_def));
 
 #define ZREAD(filefunc,filestream,buf,size) ((*((filefunc).zread_file))((filefunc).opaque,filestream,buf,size))
 #define ZWRITE(filefunc,filestream,buf,size) ((*((filefunc).zwrite_file))((filefunc).opaque,filestream,buf,size))
diff -Nrub qmc2/minizip/unzip.c qmc2-new/minizip/unzip.c
--- qmc2/minizip/unzip.c	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/unzip.c	2011-11-13 22:20:10.000000000 -0500
@@ -163,7 +163,7 @@
 */
 
 
-local int unzlocal_getByte OF((
+local int unzlocal_getByte _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     int *pi));
@@ -193,7 +193,7 @@
 /* ===========================================================================
    Reads a long in LSB order from the given gz_stream. Sets
 */
-local int unzlocal_getShort OF((
+local int unzlocal_getShort _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     uLong *pX));
@@ -221,7 +221,7 @@
     return err;
 }
 
-local int unzlocal_getLong OF((
+local int unzlocal_getLong _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     uLong *pX));
@@ -324,7 +324,7 @@
   Locate the Central directory of a zipfile (at the end, just before
     the global comment)
 */
-local uLong unzlocal_SearchCentralDir OF((
+local uLong unzlocal_SearchCentralDir _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream));
 
@@ -563,7 +563,7 @@
 /*
   Get Info about the current file in the zipfile, with internal only info
 */
-local int unzlocal_GetCurrentFileInfoInternal OF((unzFile file,
+local int unzlocal_GetCurrentFileInfoInternal _Z_OF((unzFile file,
                                                   unz_file_info *pfile_info,
                                                   unz_file_info_internal
                                                   *pfile_info_internal,
diff -Nrub qmc2/minizip/unzip.h qmc2-new/minizip/unzip.h
--- qmc2/minizip/unzip.h	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/unzip.h	2011-11-13 22:20:56.000000000 -0500
@@ -119,7 +119,7 @@
     tm_unz tmu_date;
 } unz_file_info;
 
-extern int ZEXPORT unzStringFileNameCompare OF ((const char* fileName1,
+extern int ZEXPORT unzStringFileNameCompare _Z_OF ((const char* fileName1,
                                                  const char* fileName2,
                                                  int iCaseSensitivity));
 /*
@@ -132,7 +132,7 @@
 */
 
 
-extern unzFile ZEXPORT unzOpen OF((const char *path));
+extern unzFile ZEXPORT unzOpen _Z_OF((const char *path));
 /*
   Open a Zip file. path contain the full pathname (by example,
      on a Windows XP computer "c:\\zlib\\zlib113.zip" or on an Unix computer
@@ -143,21 +143,21 @@
        of this unzip package.
 */
 
-extern unzFile ZEXPORT unzOpen2 OF((const char *path,
+extern unzFile ZEXPORT unzOpen2 _Z_OF((const char *path,
                                     zlib_filefunc_def* pzlib_filefunc_def));
 /*
    Open a Zip file, like unzOpen, but provide a set of file low level API
       for read/write the zip file (see ioapi.h)
 */
 
-extern int ZEXPORT unzClose OF((unzFile file));
+extern int ZEXPORT unzClose _Z_OF((unzFile file));
 /*
   Close a ZipFile opened with unzipOpen.
   If there is files inside the .Zip opened with unzOpenCurrentFile (see later),
     these files MUST be closed with unzipCloseCurrentFile before call unzipClose.
   return UNZ_OK if there is no problem. */
 
-extern int ZEXPORT unzGetGlobalInfo OF((unzFile file,
+extern int ZEXPORT unzGetGlobalInfo _Z_OF((unzFile file,
                                         unz_global_info *pglobal_info));
 /*
   Write info about the ZipFile in the *pglobal_info structure.
@@ -165,7 +165,7 @@
   return UNZ_OK if there is no problem. */
 
 
-extern int ZEXPORT unzGetGlobalComment OF((unzFile file,
+extern int ZEXPORT unzGetGlobalComment _Z_OF((unzFile file,
                                            char *szComment,
                                            uLong uSizeBuf));
 /*
@@ -178,20 +178,20 @@
 /***************************************************************************/
 /* Unzip package allow you browse the directory of the zipfile */
 
-extern int ZEXPORT unzGoToFirstFile OF((unzFile file));
+extern int ZEXPORT unzGoToFirstFile _Z_OF((unzFile file));
 /*
   Set the current file of the zipfile to the first file.
   return UNZ_OK if there is no problem
 */
 
-extern int ZEXPORT unzGoToNextFile OF((unzFile file));
+extern int ZEXPORT unzGoToNextFile _Z_OF((unzFile file));
 /*
   Set the current file of the zipfile to the next file.
   return UNZ_OK if there is no problem
   return UNZ_END_OF_LIST_OF_FILE if the actual file was the latest.
 */
 
-extern int ZEXPORT unzLocateFile OF((unzFile file,
+extern int ZEXPORT unzLocateFile _Z_OF((unzFile file,
                      const char *szFileName,
                      int iCaseSensitivity));
 /*
@@ -223,7 +223,7 @@
 
 /* ****************************************** */
 
-extern int ZEXPORT unzGetCurrentFileInfo OF((unzFile file,
+extern int ZEXPORT unzGetCurrentFileInfo _Z_OF((unzFile file,
                          unz_file_info *pfile_info,
                          char *szFileName,
                          uLong fileNameBufferSize,
@@ -249,13 +249,13 @@
    from it, and close it (you can close it before reading all the file)
    */
 
-extern int ZEXPORT unzOpenCurrentFile OF((unzFile file));
+extern int ZEXPORT unzOpenCurrentFile _Z_OF((unzFile file));
 /*
   Open for reading data the current file in the zipfile.
   If there is no error, the return value is UNZ_OK.
 */
 
-extern int ZEXPORT unzOpenCurrentFilePassword OF((unzFile file,
+extern int ZEXPORT unzOpenCurrentFilePassword _Z_OF((unzFile file,
                                                   const char* password));
 /*
   Open for reading data the current file in the zipfile.
@@ -263,7 +263,7 @@
   If there is no error, the return value is UNZ_OK.
 */
 
-extern int ZEXPORT unzOpenCurrentFile2 OF((unzFile file,
+extern int ZEXPORT unzOpenCurrentFile2 _Z_OF((unzFile file,
                                            int* method,
                                            int* level,
                                            int raw));
@@ -276,7 +276,7 @@
          but you CANNOT set method parameter as NULL
 */
 
-extern int ZEXPORT unzOpenCurrentFile3 OF((unzFile file,
+extern int ZEXPORT unzOpenCurrentFile3 _Z_OF((unzFile file,
                                            int* method,
                                            int* level,
                                            int raw,
@@ -291,13 +291,13 @@
 */
 
 
-extern int ZEXPORT unzCloseCurrentFile OF((unzFile file));
+extern int ZEXPORT unzCloseCurrentFile _Z_OF((unzFile file));
 /*
   Close the file in zip opened with unzOpenCurrentFile
   Return UNZ_CRCERROR if all the file was read but the CRC is not good
 */
 
-extern int ZEXPORT unzReadCurrentFile OF((unzFile file,
+extern int ZEXPORT unzReadCurrentFile _Z_OF((unzFile file,
                       voidp buf,
                       unsigned len));
 /*
@@ -311,17 +311,17 @@
     (UNZ_ERRNO for IO error, or zLib error for uncompress error)
 */
 
-extern z_off_t ZEXPORT unztell OF((unzFile file));
+extern z_off_t ZEXPORT unztell _Z_OF((unzFile file));
 /*
   Give the current position in uncompressed data
 */
 
-extern int ZEXPORT unzeof OF((unzFile file));
+extern int ZEXPORT unzeof _Z_OF((unzFile file));
 /*
   return 1 if the end of file was reached, 0 elsewhere
 */
 
-extern int ZEXPORT unzGetLocalExtrafield OF((unzFile file,
+extern int ZEXPORT unzGetLocalExtrafield _Z_OF((unzFile file,
                                              voidp buf,
                                              unsigned len));
 /*
diff -Nrub qmc2/minizip/zip.c qmc2-new/minizip/zip.c
--- qmc2/minizip/zip.c	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/zip.c	2011-11-13 22:20:10.000000000 -0500
@@ -261,7 +261,7 @@
    nbByte == 1, 2 or 4 (byte, short or long)
 */
 
-local int ziplocal_putValue OF((const zlib_filefunc_def* pzlib_filefunc_def,
+local int ziplocal_putValue _Z_OF((const zlib_filefunc_def* pzlib_filefunc_def,
                                 voidpf filestream, uLong x, int nbByte));
 local int ziplocal_putValue (pzlib_filefunc_def, filestream, x, nbByte)
     const zlib_filefunc_def* pzlib_filefunc_def;
@@ -290,7 +290,7 @@
         return ZIP_OK;
 }
 
-local void ziplocal_putValue_inmemory OF((void* dest, uLong x, int nbByte));
+local void ziplocal_putValue_inmemory _Z_OF((void* dest, uLong x, int nbByte));
 local void ziplocal_putValue_inmemory (dest, x, nbByte)
     void* dest;
     uLong x;
@@ -332,7 +332,7 @@
 
 /****************************************************************************/
 
-local int ziplocal_getByte OF((
+local int ziplocal_getByte _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     int *pi));
@@ -362,7 +362,7 @@
 /* ===========================================================================
    Reads a long in LSB order from the given gz_stream. Sets
 */
-local int ziplocal_getShort OF((
+local int ziplocal_getShort _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     uLong *pX));
@@ -390,7 +390,7 @@
     return err;
 }
 
-local int ziplocal_getLong OF((
+local int ziplocal_getLong _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream,
     uLong *pX));
@@ -433,7 +433,7 @@
   Locate the Central directory of a zipfile (at the end, just before
     the global comment)
 */
-local uLong ziplocal_SearchCentralDir OF((
+local uLong ziplocal_SearchCentralDir _Z_OF((
     const zlib_filefunc_def* pzlib_filefunc_def,
     voidpf filestream));
 
diff -Nrub qmc2/minizip/zip.h qmc2-new/minizip/zip.h
--- qmc2/minizip/zip.h	2011-05-12 12:58:49.000000000 -0400
+++ qmc2-new/minizip/zip.h	2011-11-13 22:20:10.000000000 -0500
@@ -111,7 +111,7 @@
 #define APPEND_STATUS_CREATEAFTER   (1)
 #define APPEND_STATUS_ADDINZIP      (2)
 
-extern zipFile ZEXPORT zipOpen OF((const char *pathname, int append));
+extern zipFile ZEXPORT zipOpen _Z_OF((const char *pathname, int append));
 /*
   Create a zipfile.
      pathname contain on Windows XP a filename like "c:\\zlib\\zlib113.zip" or on
@@ -131,12 +131,12 @@
    Of couse, you can use RAW reading and writing to copy the file you did not want delte
 */
 
-extern zipFile ZEXPORT zipOpen2 OF((const char *pathname,
+extern zipFile ZEXPORT zipOpen2 _Z_OF((const char *pathname,
                                    int append,
                                    zipcharpc* globalcomment,
                                    zlib_filefunc_def* pzlib_filefunc_def));
 
-extern int ZEXPORT zipOpenNewFileInZip OF((zipFile file,
+extern int ZEXPORT zipOpenNewFileInZip _Z_OF((zipFile file,
                        const char* filename,
                        const zip_fileinfo* zipfi,
                        const void* extrafield_local,
@@ -160,7 +160,7 @@
 */
 
 
-extern int ZEXPORT zipOpenNewFileInZip2 OF((zipFile file,
+extern int ZEXPORT zipOpenNewFileInZip2 _Z_OF((zipFile file,
                                             const char* filename,
                                             const zip_fileinfo* zipfi,
                                             const void* extrafield_local,
@@ -176,7 +176,7 @@
   Same than zipOpenNewFileInZip, except if raw=1, we write raw file
  */
 
-extern int ZEXPORT zipOpenNewFileInZip3 OF((zipFile file,
+extern int ZEXPORT zipOpenNewFileInZip3 _Z_OF((zipFile file,
                                             const char* filename,
                                             const zip_fileinfo* zipfi,
                                             const void* extrafield_local,
@@ -201,19 +201,19 @@
  */
 
 
-extern int ZEXPORT zipWriteInFileInZip OF((zipFile file,
+extern int ZEXPORT zipWriteInFileInZip _Z_OF((zipFile file,
                        const void* buf,
                        unsigned len));
 /*
   Write data in the zipfile
 */
 
-extern int ZEXPORT zipCloseFileInZip OF((zipFile file));
+extern int ZEXPORT zipCloseFileInZip _Z_OF((zipFile file));
 /*
   Close the current file in the zipfile
 */
 
-extern int ZEXPORT zipCloseFileInZipRaw OF((zipFile file,
+extern int ZEXPORT zipCloseFileInZipRaw _Z_OF((zipFile file,
                                             uLong uncompressed_size,
                                             uLong crc32));
 /*
@@ -222,7 +222,7 @@
   uncompressed_size and crc32 are value for the uncompressed size
 */
 
-extern int ZEXPORT zipClose OF((zipFile file,
+extern int ZEXPORT zipClose _Z_OF((zipFile file,
                 const char* global_comment));
 /*
   Close the zipfile
