diff --git a/Makefile b/Makefile
index 5c79d41..f6d0967 100644
--- a/Makefile
+++ b/Makefile
@@ -828,10 +828,9 @@ endif
 	  $(SED) -e "s*DATADIR*$(DATADIR)*g" < ./inst/$(PROJECT).ini.template > "$(GLOBAL_QMC2_INI)" ; \
 	fi
 ifneq '$(ARCH)' 'Darwin'
-	@echo "Installing $(TARGET_NAME).desktop to $(GLOBAL_DATADIR)/applications"
-	@$(MKDIR) $(GLOBAL_DATADIR)/applications
-	@$(CHMOD) a+rx $(GLOBAL_DATADIR)/applications
-	@$(SED) -e "s*DATADIR*$(DATADIR)*g; s*EMULATOR*$(EMULATOR)*g; s*TARGET*$(TARGET_NAME)*g; s*EMUICO*$(EMUICO)*g; s*GENERICNAME*$(GENERICNAME)*g" < ./inst/$(PROJECT).desktop.template > $(GLOBAL_DATADIR)/applications/$(TARGET_NAME).desktop
+	@echo "Installing $(TARGET_NAME).desktop to $(DESTDIR)/usr/share/applications"
+	@$(MKDIR) $(DESTDIR)/share/applications
+	@$(SED) -e "s*DATADIR*$(DATADIR)*g; s*EMULATOR*$(EMULATOR)*g; s*TARGET*$(TARGET_NAME)*g; s*EMUICO*$(EMUICO)*g; s*GENERICNAME*$(GENERICNAME)*g" < ./inst/$(PROJECT).desktop.template > $(DESTDIR)/share/applications/$(TARGET_NAME).desktop
 endif
 	@echo "Installation complete"
 
