diff --git a/snesfilter/Makefile b/snesfilter/Makefile
index 089b86e..08ba81b 100755
--- a/snesfilter/Makefile
+++ b/snesfilter/Makefile
@@ -5,12 +5,12 @@ include nall/qt/Makefile
 
 c     := $(compiler) -x c -std=gnu99
 cpp   := $(compiler) -std=gnu++0x
-flags := -O3 -I. -Iobj -fomit-frame-pointer $(qtinc)
+flags := $(CXXFLAGS) -I. -Iobj -fomit-frame-pointer $(qtinc)
 link  :=
 
 ifeq ($(platform),x)
   flags := -fPIC -fopenmp $(flags)
-  link += -s -fopenmp -lpthread -lgomp
+  link += -fopenmp -lpthread -lgomp
 else ifeq ($(platform),osx)
   flags := -fPIC -fopenmp $(flags)
   link += -fopenmp -lpthread -lgomp
