Index: phonetooth/contactsdialog.py
===================================================================
--- phonetooth/contactsdialog.py	(wersja 150)
+++ phonetooth/contactsdialog.py	(kopia robocza)
@@ -124,13 +124,13 @@
             
             gtk.gdk.threads_enter()
             self.__updateStoreFromContactList()
+            self.__setSensitive(True)
             gtk.gdk.threads_leave()
         except Exception, e:
             gtk.gdk.threads_enter()
             self.__error(str(e))
+            self.__setSensitive(True)
             gtk.gdk.threads_leave()
-        finally:
-            gobject.idle_add(self.__setSensitive, True)
         
         
     def __exportContacts(self, widget):
Index: phonetooth/preferencesdialog.py
===================================================================
--- phonetooth/preferencesdialog.py	(wersja 150)
+++ phonetooth/preferencesdialog.py	(kopia robocza)
@@ -100,6 +100,7 @@
             devices = discoverer.findSerialDevices()
             gtk.gdk.threads_enter()
             self.__setDevices(devices)
+            self.__setWaiting(False)
             gtk.gdk.threads_leave()
         except Exception, e:
             gtk.gdk.threads_enter()
@@ -107,9 +108,6 @@
             errorDlg.set_title(_('Error'))
             errorDlg.run()
             errorDlg.destroy()
-            gtk.gdk.threads_leave()
-        finally:
-            gtk.gdk.threads_enter()
             self.__setWaiting(False)
             gtk.gdk.threads_leave()
         
