diff --git a/Makefile.in b/Makefile.in
index c8b1e1a..6974910 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -64,7 +64,6 @@ DIST_COMMON = README $(am__configure_deps) $(dist_doc_DATA) \
 	$(top_srcdir)/config_aux/install-sh \
 	$(top_srcdir)/config_aux/ltmain.sh \
 	$(top_srcdir)/config_aux/missing $(top_srcdir)/configure \
-	ABOUT-NLS AUTHORS COPYING ChangeLog INSTALL NEWS \
 	config_aux/config.guess config_aux/config.rpath \
 	config_aux/config.sub config_aux/depcomp config_aux/install-sh \
 	config_aux/ltmain.sh config_aux/missing
@@ -387,12 +386,7 @@ pushover_SOURCES = \
   src/tools.cpp src/tools.h \
   src/window.cpp src/window.h
 
-dist_doc_DATA = \
-  README \
-  AUTHORS \
-  COPYING \
-  NEWS \
-  ChangeLog
+dist_doc_DATA =
 
 nobase_dist_pkgdata_DATA = \
   data/ant.png \
diff --git a/configure b/configure
index ade7715..d9e7a1a 100755
--- a/configure
+++ b/configure
@@ -15302,7 +15302,7 @@ $as_echo_n "checking whether NLS is requested... " >&6; }
 if test "${enable_nls+set}" = set; then :
   enableval=$enable_nls; USE_NLS=$enableval
 else
-  USE_NLS=yes
+  USE_NLS=no
 fi
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $USE_NLS" >&5
