--- ../i386/cpuexec.S	2009-09-04 05:26:00.000000000 -0400
+++ ../i386/cpuexec.S	2009-09-04 05:28:02.000000000 -0400
@@ -709,3 +709,7 @@
 	andl $~IRQ_PENDING_FLAG, Flags	
 .irqsstillpending:
 	ret
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/sa1gs.S	2009-09-04 05:26:12.000000000 -0400
+++ ../i386/sa1gs.S	2009-09-04 05:28:30.000000000 -0400
@@ -591,3 +591,7 @@
 	addl %eax, %edx
 	movl %edx, PC
 	ret
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/cpuops.S	2009-09-04 05:26:22.000000000 -0400
+++ ../i386/cpuops.S	2009-09-04 05:28:55.000000000 -0400
@@ -4735,3 +4735,7 @@
 	.long OpFDM0
 	.long OpFEM0
 	.long OpFFM0
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/getset.S	2009-09-04 05:26:37.000000000 -0400
+++ ../i386/getset.S	2009-09-04 05:29:12.000000000 -0400
@@ -1271,3 +1271,7 @@
 .SPC7110NULL:
         movl %eax, PCBase
 	ret
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/spc700.S	2009-09-04 05:26:48.000000000 -0400
+++ ../i386/spc700.S	2009-09-04 05:29:38.000000000 -0400
@@ -7570,3 +7570,7 @@
 	.long ApuOpFD
 	.long ApuOpFE
 	.long ApuOpFF
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/sa1ops.S	2009-09-04 05:26:58.000000000 -0400
+++ ../i386/sa1ops.S	2009-09-04 05:29:57.000000000 -0400
@@ -4617,3 +4617,7 @@
 	.long OpFDM0
 	.long OpFEM0
 	.long OpFFM0
+
+#if defined(__linux__) && defined(__ELF__)
+.section .note.GNU-stack,"",%progbits
+#endif
--- ../i386/SPC.ASM	2009-09-04 07:01:26.000000000 -0400
+++ ../i386/SPC.ASM	2009-09-04 07:03:21.000000000 -0400
@@ -513,3 +513,7 @@
 	mov edx, 32767
 CLIP16SMP2SKIP1:
 	ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/ZSNES.ASM	2009-09-04 07:01:58.000000000 -0400
+++ ../i386/ZSNES.ASM	2009-09-04 07:03:47.000000000 -0400
@@ -374,3 +374,7 @@
    call UpdatePORSCMR
    call UpdateCLSR
    ret
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/C4.ASM	2009-09-04 07:01:12.000000000 -0400
+++ ../i386/C4.ASM	2009-09-04 07:02:56.000000000 -0400
@@ -2388,3 +2388,7 @@
 dw $07FE9,$07FF6,$07FFD
 
 SECTION .text
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/2XSAIMMX.ASM	2009-09-04 07:01:16.000000000 -0400
+++ ../i386/2XSAIMMX.ASM	2009-09-04 07:02:26.000000000 -0400
@@ -2059,3 +2059,7 @@
 final1b       resb 8
 final2a       resb 8
 final2b       resb 8
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/sfxproc.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/sfxproc.asm	2009-09-04 07:10:53.000000000 -0400
@@ -676,3 +676,7 @@
     ret
 
 NEWSYM SfxProcAsmEnd
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/bilinear.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/bilinear.asm	2009-09-04 07:08:44.000000000 -0400
@@ -412,3 +412,8 @@
 BlueMask      dd 0x001F001F, 0x001F001F
 GreenMask     dd 0x07E007E0, 0x07E007E0
 All32s        dd 0x00200020, 0x00200020
+
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/fxemu2.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/fxemu2.asm	2009-09-04 07:09:13.000000000 -0400
@@ -2690,3 +2690,7 @@
 SECTION .text
 
 NEWSYM FxEmu2AsmEnd
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/fxemu2c.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/fxemu2c.asm	2009-09-04 07:10:03.000000000 -0400
@@ -2555,3 +2555,7 @@
    FXReturn
 
 NEWSYM FxEmu2CAsmEnd
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/zsnesc4.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/zsnesc4.asm	2009-09-04 07:11:10.000000000 -0400
@@ -90,3 +90,7 @@
 	and ecx, 0ffffh
 	sub ecx, 06000h
 	jmp C4ReadReg
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/fxemu2b.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/fxemu2b.asm	2009-09-04 07:09:38.000000000 -0400
@@ -615,3 +615,7 @@
    ret
 
 NEWSYM FxEmu2BAsmEnd
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
--- ../i386/fxtable.asm	2009-09-04 07:07:17.000000000 -0400
+++ ../i386/fxtable.asm	2009-09-04 07:10:21.000000000 -0400
@@ -3555,3 +3555,7 @@
 
 SECTION .text
 NEWSYM FxTableAsmEnd
+
+%ifidn __OUTPUT_FORMAT__,elf
+section .note.GNU-stack noalloc noexec nowrite progbits
+%endif
