diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ApplicationUsage OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ApplicationUsage
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ApplicationUsage	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ApplicationUsage	2009-03-17 16:57:19.918536457 +0100
@@ -19,6 +19,7 @@
 #include <map>
 #include <string>
 #include <ostream>
+#include <cstdlib>
 
 namespace osg {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ArgumentParser OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ArgumentParser
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ArgumentParser	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ArgumentParser	2009-03-17 17:01:47.334536167 +0100
@@ -17,8 +17,9 @@
 #include <osg/Export>
 
 #include <map>
-#include <string>
+#include <cstring>
 #include <ostream>
+#include <cstdlib>
 
 namespace osg {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendColor OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendColor
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendColor	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendColor	2009-03-17 17:00:19.042536168 +0100
@@ -19,7 +19,7 @@
 #include <osg/ref_ptr>
 #include <osg/Vec4>
 
-
+#include <cstring>
 
 namespace osg {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendEquation OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendEquation
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendEquation	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendEquation	2009-03-17 17:00:35.562536165 +0100
@@ -16,6 +16,8 @@
 
 #include <osg/StateAttribute>
 
+#include <cstring>
+
 #ifndef GL_VERSION_1_2
 /* Logic Ops */
 #define GL_MIN                                  0x8007
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendFunc OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendFunc
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BlendFunc	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BlendFunc	2009-03-17 17:00:43.898536156 +0100
@@ -16,6 +16,7 @@
 
 #include <osg/StateAttribute>
 
+#include <cstring>
 
 #ifndef GL_VERSION_1_2
 #define GL_CONSTANT_COLOR                 0x8001
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BufferObject OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BufferObject
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/BufferObject	2006-08-25 10:48:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/BufferObject	2009-03-17 17:04:01.070536554 +0100
@@ -17,6 +17,7 @@
 #include <osg/GL>
 #include <osg/Object>
 #include <osg/buffered_value>
+#include <cstring>
 
 #ifndef GL_ARB_vertex_buffer_object
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ClampColor OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ClampColor
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/ClampColor	2006-09-12 11:26:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/ClampColor	2009-03-17 17:08:24.334536199 +0100
@@ -16,6 +16,8 @@
 
 #include <osg/StateAttribute>
 
+#include <cstring>
+
 #ifndef GL_ARB_color_buffer_float
 #define GL_RGBA_FLOAT_MODE_ARB                  0x8820
 #define GL_CLAMP_VERTEX_COLOR_ARB               0x891A
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Geode OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Geode
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Geode	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Geode	2009-03-17 17:44:19.662536282 +0100
@@ -18,6 +18,8 @@
 #include <osg/NodeVisitor>
 #include <osg/Drawable>
 
+#include <iostream>
+
 namespace osg {
 
 /** A \c Geode is a "geometry node", that is, a leaf node on the scene graph
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Geometry OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Geometry
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Geometry	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Geometry	2009-03-17 17:11:20.342536159 +0100
@@ -21,6 +21,8 @@
 #include <osg/Array>
 #include <osg/PrimitiveSet>
 
+#include <cstdlib>
+
 namespace osg {
 
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Notify OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Notify
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Notify	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Notify	2009-03-17 17:13:06.786536158 +0100
@@ -17,6 +17,7 @@
 #include <osg/Export>
 
 #include <ostream>
+#include <cstdlib>
 
 namespace osg {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Point OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Point
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Point	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Point	2009-03-17 17:18:51.802536356 +0100
@@ -17,6 +17,8 @@
 #include <osg/Vec3>
 #include <osg/StateAttribute>
 
+#include <cstring>
+
 namespace osg {
 
 /** Point - encapsulates the OpenGL point smoothing and size state.*/
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/PolygonOffset OSG_OP_OT-1.2/OpenSceneGraph/include/osg/PolygonOffset
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/PolygonOffset	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/PolygonOffset	2009-03-17 17:20:37.002536563 +0100
@@ -16,6 +16,8 @@
 
 #include <osg/StateAttribute>
 
+#include <cstring>
+
 namespace osg {
 
 /** PolygonOffset - encapsulates the OpenGL glPolygonOffset state.*/
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/TexEnvFilter OSG_OP_OT-1.2/OpenSceneGraph/include/osg/TexEnvFilter
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/TexEnvFilter	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/TexEnvFilter	2009-03-17 17:22:40.462536606 +0100
@@ -23,6 +23,8 @@
 #define GL_TEXTURE_LOD_BIAS_EXT           0x8501
 #endif
 
+#include <cstdlib>
+
 namespace osg {
 
 /** TexEnvFilter - encapsulates the OpenGL glTexEnv (GL_TEXTURE_FILTER_CONTROL) state.*/
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Uniform OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Uniform
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osg/Uniform	2006-05-16 23:00:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osg/Uniform	2009-03-17 17:25:19.478536156 +0100
@@ -25,7 +25,7 @@
 #include <osg/Matrix>
 #include <osg/GL2Extensions>
 
-#include <string>
+#include <cstring>
 #include <vector>
 
 namespace osg {
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/Field OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/Field
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/Field	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/Field	2009-03-17 17:39:38.786536720 +0100
@@ -18,6 +18,7 @@
 
 #include <string>
 #include <stdlib.h>
+#include <cstring>
 
 namespace osgDB {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/FieldReader OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/FieldReader
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/FieldReader	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/FieldReader	2009-03-17 17:36:31.014536449 +0100
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 
 #include <istream>
+#include <cstring>
 
 namespace osgDB {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/FileNameUtils OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/FileNameUtils
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/include/osgDB/FileNameUtils	2006-07-18 17:21:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/include/osgDB/FileNameUtils	2009-03-17 17:33:04.002574369 +0100
@@ -17,6 +17,7 @@
 #include <osgDB/Export>
 
 #include <string>
+#include <cstring>
 
 namespace osgDB {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/directx/directx.h OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/directx/directx.h
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/directx/directx.h	2006-07-04 11:13:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/directx/directx.h	2009-03-17 17:53:16.986536694 +0100
@@ -27,6 +27,7 @@
 #include <string>
 #include <vector>
 #include <fstream>
+#include <cstring>
 
 #include <osg/Export>
 #include <osg/Math>
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/directx/types.h OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/directx/types.h
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/directx/types.h	2006-07-05 10:59:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/directx/types.h	2009-03-17 17:53:06.694536462 +0100
@@ -27,6 +27,7 @@
 #include <string>
 #include <vector>
 #include <fstream>
+#include <cstdlib>
 
 #include <osg/Math>
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/lwo/ReaderWriterLWO.cpp OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/lwo/ReaderWriterLWO.cpp
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/lwo/ReaderWriterLWO.cpp	2006-08-02 12:43:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/lwo/ReaderWriterLWO.cpp	2009-03-17 17:59:32.318536258 +0100
@@ -18,6 +18,7 @@
 #include <string>
 #include <sstream>
 #include <algorithm>
+#include <memory>
 
 #include <osg/Notify>
 #include <osg/Node>
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/OpenFlight/DataInputStream.h OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/OpenFlight/DataInputStream.h
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/OpenFlight/DataInputStream.h	2006-06-29 11:34:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/OpenFlight/DataInputStream.h	2009-03-17 17:55:52.130536250 +0100
@@ -14,6 +14,7 @@
 #include <osg/Vec3d>
 #include <osg/Vec4f>
 #include "types.h"    
+#include <cstring>
 
 namespace flt {
 
diff -Nur OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/txp/trpage_swap.h OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/txp/trpage_swap.h
--- OSG_OP_OT-1.2.orig/OpenSceneGraph/src/osgPlugins/txp/trpage_swap.h	2006-06-27 12:07:00.000000000 +0200
+++ OSG_OP_OT-1.2/OpenSceneGraph/src/osgPlugins/txp/trpage_swap.h	2009-03-17 18:04:52.566536154 +0100
@@ -24,6 +24,8 @@
 
 #include <trpage_io.h>
 
+#include <cstring>
+
 // Byte swap and return a short
 // {group:Byte Ordering Utilities}
 short trpg_byteswap_short( short number );
