diff --git a/plugins/spellchecker/configuration/spellchecker-configuration.cpp b/plugins/spellchecker/configuration/spellchecker-configuration.cpp
index bd74f54..e7d315d 100644
--- a/plugins/spellchecker/configuration/spellchecker-configuration.cpp
+++ b/plugins/spellchecker/configuration/spellchecker-configuration.cpp
@@ -20,8 +20,8 @@
 #include "configuration/configuration-file.h"
 
 #include "spellchecker-configuration.h"
-#include "plugins/spellchecker/spellchecker.h"
-#include "plugins/spellchecker/spellchecker-plugin.h"
+#include "spellchecker.h" 
+#include "spellchecker-plugin.h"
 
 SpellcheckerConfiguration *SpellcheckerConfiguration::Instance = 0;
 
