diff -Nur qcdemu.orig/qcdemu.pro qcdemu/qcdemu.pro
--- qcdemu.orig/qcdemu.pro	2008-08-06 11:22:41.000000000 +0200
+++ qcdemu/qcdemu.pro	2009-03-06 18:59:20.431006880 +0100
@@ -38,7 +38,7 @@
 translations.path = $${TRANSLATION_PATH}
 
 mime.files = mime/application/*.desktop
-mime.path = $${INSTALL_ROOT}/$(KDEDIR)/share/mimelnk/application
+mime.path = $${INSTALL_ROOT}/$${INSTALL_PREFIX}/share/mimelnk/application
 
 applnk.files = $${TARGET}.desktop
 applnk.path = $${INSTALL_PREFIX}/share/applnk/Multimedia
