diff -Nur jrpg.orig/jrpg.py jrpg/jrpg.py
--- jrpg.orig/jrpg.py	2007-06-28 11:48:13.000000000 +0200
+++ jrpg/jrpg.py	2009-05-05 17:42:18.489414536 +0200
@@ -550,7 +550,7 @@
         # Verify that the xp in the savefile is not higher than your XP
         xp = 0
         try:
-            f = open("savefile", "r")
+            f = open(util.savedir+"savefile", "r")
             ld = pickle.load(f)
             xp = ld["xp"]
         except IOError, (errno, strerror):
@@ -561,7 +561,8 @@
                             U"If you want to save anyway, press F2 again"], (255,0,0))
             self.save_warned = True
             return
-        f = open("savefile", "w")
+        util.check_savedir()
+        f = open(util.savedir+"savefile", "w")
         save_data = {
             "name"     : "Freya",
             "hpmax"    : self.hpmax,
@@ -590,7 +591,7 @@
             self.make_load_warning = False
             return
         try:
-            f = open("savefile", "r")
+            f = open(util.savedir+"savefile", "r")
             ld = pickle.load(f)
             self.hpmax     = ld["hpmax"]
             self.hp        = ld["hp"]
@@ -2608,9 +2609,10 @@
 class Mistakes:
     def __init__(self):
         y,mo,d,h,mi,s,wd,yd,isdst = time.localtime()
+        util.check_savedir()
         fn = "mistakes-%04d-%02d-%02d-%02d-%02d-%02d.txt" % (y,mo,d,h,mi,s)
         #self.fh = open(fn, "ab")
-        self.fh = codecs.open(fn, mode="ab", encoding='utf-8')
+        self.fh = codecs.open(util.savedir+fn, mode="ab", encoding='utf-8')
     def mistake(self, attack, soul):
         msg = "Mistake: tried (%s) on demon (%s)\n" % (attack, soul.xp_code())
         self.fh.write(msg)
diff -Nur jrpg.orig/util.py jrpg/util.py
--- jrpg.orig/util.py	2007-06-06 03:46:02.000000000 +0200
+++ jrpg/util.py	2009-05-05 17:39:49.997414819 +0200
@@ -6,6 +6,7 @@
 import pygame
 import traceback
 import time
+import os
 
 try:
     import posix
@@ -34,6 +35,9 @@
     "kochi-gothic-subst.ttf",
 ]
 
+# Directory where the savefile and logs are stored
+savedir = os.path.expandvars("$HOME/.jrpg/")
+
 ############################################################
 # This class manages notifications                         #
 # It's used mostly together with Cached object             #
@@ -96,11 +100,17 @@
         return 0
     else:
         return -1
+
+def check_savedir():
+    # Create savedir if needed
+    if not os.path.isdir(savedir):
+        os.mkdir(savedir)
     
 def save_errormsg(trace_back):
     (tp,v,tb) = trace_back
     tbf = traceback.format_exception(tp,v,tb)
-    f = open("errormsg.txt", "a")
+    check_savedir()
+    f = open(savedir+"errormsg.txt", "a")
     f.write("== ")
     f.write(time.asctime(time.localtime()))
     f.write(" ==\n")
