--- tenebrae_0/sv_main.c.unpatched	2005-01-10 16:22:13.668439520 -0500
+++ tenebrae_0/sv_main.c	2005-01-10 16:22:26.761449080 -0500
@@ -1257,6 +1257,9 @@ void SV_SpawnServer (char *server)
 	
 	ED_LoadFromFile (sv.worldmodel->entities);
 
+	// Zero out the angles vectors for the world model. This fixes e3m3.
+	sv.edicts->v.angles[0]=sv.edicts->v.angles[1]=sv.edicts->v.angles[2]=0;
+
 	sv.active = true;
 
 // all setup is completed, any further precache statements are errors
