diff -Naur tenebrae_0.old/cl_main.c tenebrae_0/cl_main.c
--- tenebrae_0.old/cl_main.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/cl_main.c	2008-04-19 23:54:46.000000000 +0200
@@ -894,4 +894,4 @@
 	Cmd_AddCommand ("stop", CL_Stop_f);
 	Cmd_AddCommand ("playdemo", CL_PlayDemo_f);
 	Cmd_AddCommand ("timedemo", CL_TimeDemo_f);
-}
\ No newline at end of file
+}
diff -Naur tenebrae_0.old/common.c tenebrae_0/common.c
--- tenebrae_0.old/common.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/common.c	2008-04-19 23:32:07.000000000 +0200
@@ -2221,7 +2221,10 @@
 		if ((userdir[j-1] == '\\') || (userdir[j-1] == '/'))
 			userdir[j-1] = 0;
 	}
+
+	strncpy (com_prefdir, userdir, j);
 	COM_AddGameFS (userdir);
+	Sys_mkdir(com_gamedir);
 #endif
 
 
diff -Naur tenebrae_0.old/common.h tenebrae_0/common.h
--- tenebrae_0.old/common.h	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/common.h	2008-04-19 23:30:58.000000000 +0200
@@ -167,7 +167,11 @@
 extern int com_filesize;
 struct cache_user_s;
 
+extern char	com_prefdir[MAX_OSPATH];
+
+#ifdef  USERPREF_DIR
 extern	char	com_gamedir[MAX_OSPATH];
+#endif
 
 void COM_WriteFile (char *filename, void *data, int len);
 int COM_OpenFile (const char *filename, int *hndl);
diff -Naur tenebrae_0.old/crc.c tenebrae_0/crc.c
--- tenebrae_0.old/crc.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/crc.c	2008-04-19 23:54:55.000000000 +0200
@@ -78,4 +78,4 @@
 unsigned short CRC_Value(unsigned short crcvalue)
 {
 	return crcvalue ^ CRC_XOR_VALUE;
-}
\ No newline at end of file
+}
diff -Naur tenebrae_0.old/gl_glare.c tenebrae_0/gl_glare.c
--- tenebrae_0.old/gl_glare.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/gl_glare.c	2008-04-19 23:55:08.000000000 +0200
@@ -309,4 +309,4 @@
 	glDisable (GL_BLEND);
 	glEnable (GL_ALPHA_TEST);
 	glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-}
\ No newline at end of file
+}
diff -Naur tenebrae_0.old/gl_screenrect.c tenebrae_0/gl_screenrect.c
--- tenebrae_0.old/gl_screenrect.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/gl_screenrect.c	2008-04-19 23:55:15.000000000 +0200
@@ -171,4 +171,4 @@
 void R_SetTotalRect() {
 	glScissor(totalRect.coords[0], totalRect.coords[1],
 		totalRect.coords[2]-totalRect.coords[0],  totalRect.coords[3]-totalRect.coords[1]);	
-}
\ No newline at end of file
+}
diff -Naur tenebrae_0.old/gl_svbsp.c tenebrae_0/gl_svbsp.c
--- tenebrae_0.old/gl_svbsp.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/gl_svbsp.c	2008-04-19 23:55:20.000000000 +0200
@@ -569,4 +569,4 @@
 			VectorCopy(v[i-1],dest[j+1]);
 		} else Con_Printf("strange stuff...");
 	}
-	*/
\ No newline at end of file
+	*/
diff -Naur tenebrae_0.old/lex_part.l tenebrae_0/lex_part.l
--- tenebrae_0.old/lex_part.l	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/lex_part.l	2008-04-19 23:55:39.000000000 +0200
@@ -258,4 +258,4 @@
 void SC_End() {
 	yy_delete_buffer( buffHandle );	
 	buffHandle = NULL;
-}
\ No newline at end of file
+}
diff -Naur tenebrae_0.old/r_local.h tenebrae_0/r_local.h
--- tenebrae_0.old/r_local.h	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/r_local.h	2008-04-19 23:55:31.000000000 +0200
@@ -313,4 +313,4 @@
 void R_SplitEntityOnNode2 (mnode_t *node);
 void R_MarkLights (dlight_t *light, int bit, mnode_t *node);
 
-#endif
\ No newline at end of file
+#endif
diff -Naur tenebrae_0.old/snd_sdl.c tenebrae_0/snd_sdl.c
--- tenebrae_0.old/snd_sdl.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/snd_sdl.c	2008-04-19 23:24:41.000000000 +0200
@@ -94,7 +94,8 @@
 	shm = &the_shm;
 	shm->splitbuffer = 0;
 	shm->samplebits = (obtained.format & 0xFF);
-        shm->speed = obtained.freq;
+//        shm->speed = obtained.freq;
+	shm->speed = desired.freq;
         shm->channels = obtained.channels;
 	shm->samples = obtained.samples*shm->channels;
 	shm->samplepos = 0;
@@ -119,3 +120,12 @@
 	}
 }
 
+/*
+==============
+SNDDMA_Submit
+Send sound to device if buffer isn't really the dma buffer
+===============
+*/
+void SNDDMA_Submit(void)
+{
+}
diff -Naur tenebrae_0.old/sys_linux.c tenebrae_0/sys_linux.c
--- tenebrae_0.old/sys_linux.c	2003-07-14 23:06:01.000000000 +0200
+++ tenebrae_0/sys_linux.c	2008-04-19 23:19:45.000000000 +0200
@@ -27,7 +27,7 @@
 // user preference directory
 
 #if defined (USERPREF_DIR)
-char *prefdir=  ".quake";
+char *prefdir=  ".tenebrae";
 #endif
 
 #if defined (BASEDIR)
