diff --git a/plugins/gadu_protocol/CMakeLists.txt b/plugins/gadu_protocol/CMakeLists.txt
index a8fee80..241c5af 100644
--- a/plugins/gadu_protocol/CMakeLists.txt
+++ b/plugins/gadu_protocol/CMakeLists.txt
@@ -1,5 +1,7 @@
 project (gadu_protocol)
 
+include (FindPkgConfig)
+
 # minimal required versions
 cmake_minimum_required (VERSION 2.8)
 set (QT_MIN_VERSION "4.7.0")
@@ -12,7 +14,7 @@ include_directories ("${KADU_INCLUDE_DIR}/kadu-core")
 
 set (LIBGADU_MIN_VERSION "1.11.0")
 
-kadu_api_directories (plugins/gadu-protocol
+kadu_api_directories (plugins/gadu_protocol
 	dcc
 	file-transfers
 	gui/widgets
