--- config.lib	(revision 14508)
+++ config.lib	(working copy)
@@ -1276,10 +1276,12 @@
 
 	if [ "$with_iconv" != "0" ]; then
 		CFLAGS="$CFLAGS -DWITH_ICONV"
-		LIBS="$LIBS -liconv"
-		if [ "$with_iconv" != "2" ]; then
-			CFLAGS="$CFLAGS -I$with_iconv/include"
-			LIBS="$LIBS -L$with_iconv/lib"
+		if [ "$link_to_iconv" = "yes" ]; then
+			LIBS="$LIBS -liconv"
+			if [ "$with_iconv" != "2" ]; then
+				CFLAGS="$CFLAGS -I$with_iconv/include"
+				LIBS="$LIBS -L$with_iconv/lib"
+			fi
 		fi
 
 		if [ "$have_broken_iconv" != "no" ]; then
