diff --git a/CorsixTH/CMakeLists.txt b/CorsixTH/CMakeLists.txt
index 05e9b71..f31e177 100755
--- a/CorsixTH/CMakeLists.txt
+++ b/CorsixTH/CMakeLists.txt
@@ -138,5 +138,5 @@ install(DIRECTORY Lua Levels DESTINATION CorsixTH PATTERN "*.svn" EXCLUDE)
 install(DIRECTORY Bitmap DESTINATION CorsixTH
         FILES_MATCHING REGEX ".*\\.(tab|pal|dat|png)$"
         PATTERN "*.svn" EXCLUDE)
-install(FILES CorsixTH.lua LICENSE.txt CorsixTH.ico DESTINATION CorsixTH )
+install(FILES CorsixTH.lua CorsixTH.ico DESTINATION CorsixTH )
 
