diff --git a/engine/common.h b/engine/common.h
index b8ab488..9d58bfe 100755
--- a/engine/common.h
+++ b/engine/common.h
@@ -21,7 +21,7 @@
 #if defined(WIN32) || defined(__linux__) || defined(DOS) || defined(_M_IX86) || defined(__os2__) || defined(_X86_)
 	#define CPU_I386
 	#define MIE_LITTLE_ENDIAN
-#elif define(__MWERKS__)
+#elif defined(__MWERKS__)
 	#define CPU_PPC		/* but can't run... */
 	#define MIE_BIG_ENDIAN
 #else
