--- data/supertuxkart_desktop.template.old	2009-01-24 11:56:01.000000000 +0100
+++ data/supertuxkart_desktop.template	2009-01-24 12:07:53.000000000 +0100
@@ -3,15 +3,15 @@
 Encoding=UTF-8
 Name=Supertuxkart
 Version=0.5
-Icon=PREFIX/share/pixmaps/supertuxkart_64.xpm
+Icon=PREFIXD/pixmaps/supertuxkart_64.xpm
 Comment=A kart racing game
 Comment[fr_FR]=Un jeu de karting
 Comment[ro_RO]=Un joc de curse cu carturi
 Comment[de_DE]=Ein Kart-Rennspiel
-Exec=PREFIX/games/supertuxkart --log=file
+Exec=PREFIX/bin/supertuxkart --log=file
 Terminal=false
 StartupNotify=false
 Type=Application
-TryExec=PREFIX/games/supertuxkart
+TryExec=PREFIX/bin/supertuxkart
 Categories=Game;ArcadeGame
 GenericName=
