diff --git a/cc-gtk.sh b/cc-gtk.sh
index 06f80c4..7694a26 100755
--- a/cc-gtk.sh
+++ b/cc-gtk.sh
@@ -1,4 +1,4 @@
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c phoenix/phoenix.cpp `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c upset.cpp -DPHOENIX_GTK
-g++-4.5 -s -o upset upset.o phoenix.o `pkg-config --libs gtk+-2.0`
+g++ -std=gnu++0x -I. ${CXXFLAGS} -c phoenix/phoenix.cpp `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
+g++ -std=gnu++0x -I. ${CXXFLAGS} -c upset.cpp -DPHOENIX_GTK
+g++ -o upset upset.o phoenix.o `pkg-config --libs gtk+-2.0`
 rm *.o
diff --git a/cc-qt.sh b/cc-qt.sh
index a43eb1e..490d24d 100755
--- a/cc-qt.sh
+++ b/cc-qt.sh
@@ -1,6 +1,6 @@
 moc -i -o phoenix/qt/qt.moc phoenix/qt/qt.moc.hpp
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c phoenix/phoenix.cpp `pkg-config --cflags QtCore QtGui` -DPHOENIX_QT
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c upset.cpp -DPHOENIX_QT
-g++-4.5 -s -o upset upset.o phoenix.o `pkg-config --libs QtCore QtGui`
+g++ -std=gnu++0x -I. ${CXXFLAGS} -c phoenix/phoenix.cpp `pkg-config --cflags QtCore QtGui` -DPHOENIX_QT
+g++ -std=gnu++0x -I. ${CXXFLAGS} -c upset.cpp -DPHOENIX_QT
+g++ -o upset upset.o phoenix.o `pkg-config --libs QtCore QtGui`
 rm *.o
 rm phoenix/qt/qt.moc
