diff -ruN miastoplusa_sms/miastoplusa_sms.cpp miastoplusa_sms.new/miastoplusa_sms.cpp
--- miastoplusa_sms/miastoplusa_sms.cpp	2007-07-11 21:54:57.000000000 +0200
+++ miastoplusa_sms.new/miastoplusa_sms.cpp	2008-02-20 23:06:27.000000000 +0100
@@ -10,7 +10,7 @@
 #include "debug.h"
 #include "modules.h"
 #include "../sms/sms.h"
-#include "ui_handler.h"
+#include "miastoplusa_sms_ui_handler.h"
 #include "smsmiastoplusagateway.h"
 
 miastoplusa_sms::UIHandler uiHandler;
diff -ruN miastoplusa_sms/miastoplusa_sms_ui_handler.cpp miastoplusa_sms.new/miastoplusa_sms_ui_handler.cpp
--- miastoplusa_sms/miastoplusa_sms_ui_handler.cpp	1970-01-01 01:00:00.000000000 +0100
+++ miastoplusa_sms.new/miastoplusa_sms_ui_handler.cpp	2008-02-20 23:06:40.000000000 +0100
@@ -0,0 +1,32 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "configuration_window_widgets.h"
+#include "debug.h"
+
+#include "miastoplusa_sms_ui_handler.h"
+
+miastoplusa_sms::UIHandler::UIHandler(QObject *parent, char *name)
+{
+	kdebugf();
+}
+
+miastoplusa_sms::UIHandler::~UIHandler()
+{
+	kdebugf();
+}
+
+void miastoplusa_sms::UIHandler::mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow)
+{
+	kdebugf();
+	ConfigLineEdit *passLineEdit = dynamic_cast<ConfigLineEdit *>(mainConfigurationWindow->widgetById("miastoplusa_sms/Password"));
+	passLineEdit->setEchoMode(QLineEdit::Password);
+	kdebugf2();
+}
+
diff -ruN miastoplusa_sms/miastoplusa_sms_ui_handler.h miastoplusa_sms.new/miastoplusa_sms_ui_handler.h
--- miastoplusa_sms/miastoplusa_sms_ui_handler.h	1970-01-01 01:00:00.000000000 +0100
+++ miastoplusa_sms.new/miastoplusa_sms_ui_handler.h	2008-02-20 23:06:57.000000000 +0100
@@ -0,0 +1,22 @@
+#ifndef MIASTOPLUSA_SMS_UI_HANDLER_H
+#define MIASTOPLUSA_SMS_UI_HANDLER_H
+
+#include "main_configuration_window.h"
+
+namespace miastoplusa_sms
+{
+	class UIHandler : public ConfigurationUiHandler
+	{
+		Q_OBJECT
+
+		public:
+			UIHandler(QObject *parent = NULL, char *name = NULL);
+			~UIHandler();
+
+		public slots:
+			virtual void mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow);
+
+	};
+};
+
+#endif // MIASTOPLUSA_SMS_UI_HANDLER_H
diff -ruN miastoplusa_sms/spec miastoplusa_sms.new/spec
--- miastoplusa_sms/spec	2007-07-11 12:51:09.000000000 +0200
+++ miastoplusa_sms.new/spec	2008-02-20 23:06:10.000000000 +0100
@@ -1,5 +1,5 @@
-MODULE_SOURCES="miastoplusa_sms.cpp smsmiastoplusagateway.cpp sendthread.cpp ui_handler.cpp"
-MODULE_MOC_HEADERS="smsmiastoplusagateway.h sendthread.h ui_handler.h"
+MODULE_SOURCES="miastoplusa_sms.cpp smsmiastoplusagateway.cpp sendthread.cpp miastoplusa_sms_ui_handler.cpp"
+MODULE_MOC_HEADERS="smsmiastoplusagateway.h sendthread.h miastoplusa_sms_ui_handler.h"
 
 MODULE_LIBS=curl
 MODULE_INCLUDES=curl/curl.h
diff -ruN miastoplusa_sms/ui_handler.cpp miastoplusa_sms.new/ui_handler.cpp
--- miastoplusa_sms/ui_handler.cpp	2007-07-11 21:54:57.000000000 +0200
+++ miastoplusa_sms.new/ui_handler.cpp	1970-01-01 01:00:00.000000000 +0100
@@ -1,32 +0,0 @@
-/***************************************************************************
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- *                                                                         *
- ***************************************************************************/
-
-#include "configuration_window_widgets.h"
-#include "debug.h"
-
-#include "ui_handler.h"
-
-miastoplusa_sms::UIHandler::UIHandler(QObject *parent, char *name)
-{
-	kdebugf();
-}
-
-miastoplusa_sms::UIHandler::~UIHandler()
-{
-	kdebugf();
-}
-
-void miastoplusa_sms::UIHandler::mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow)
-{
-	kdebugf();
-	ConfigLineEdit *passLineEdit = dynamic_cast<ConfigLineEdit *>(mainConfigurationWindow->widgetById("miastoplusa_sms/Password"));
-	passLineEdit->setEchoMode(QLineEdit::Password);
-	kdebugf2();
-}
-
diff -ruN miastoplusa_sms/ui_handler.h miastoplusa_sms.new/ui_handler.h
--- miastoplusa_sms/ui_handler.h	2007-07-11 12:51:09.000000000 +0200
+++ miastoplusa_sms.new/ui_handler.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,23 +0,0 @@
-#ifndef UI_HANDLER_H
-#define UI_HANDLER_H
-
-#include "main_configuration_window.h"
-
-namespace miastoplusa_sms
-{
-	class UIHandler : public ConfigurationUiHandler
-	{
-		Q_OBJECT
-
-		public:
-			UIHandler(QObject *parent = NULL, char *name = NULL);
-			~UIHandler();
-
-		public slots:
-			virtual void mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow);
-
-	};
-};
-
-#endif // UI_HANDLER_H
-
