diff -Nur d3lphin-0.9.2orig/src/dolphinview.cpp d3lphin-0.9.2.7zip/src/dolphinview.cpp
--- d3lphin-0.9.2.old/src/dolphinview.cpp	2007-07-24 20:11:39.000000000 +0200
+++ d3lphin-0.9.2/src/dolphinview.cpp	2008-03-09 02:35:09.429521979 +0100
@@ -713,6 +713,11 @@
            url.setProtocol("zip");
            setURL(url);
        }
+       else if (mime->is("application/x-7z")) {
+           KURL url = fileItem->url();
+           url.setProtocol("p7zip");
+           setURL(url);
+       }
        else if (mime->is("application/x-tar") ||
                 mime->is("application/x-tarz") ||
                 mime->is("application/x-tbz") ||
diff -Nur d3lphin-0.9.2orig/src/urlnavigator.cpp d3lphin-0.9.2.7zip/src/urlnavigator.cpp
--- d3lphin-0.9.2.old/src/urlnavigator.cpp	2007-07-24 20:00:42.000000000 +0200
+++ d3lphin-0.9.2.7zip/src/urlnavigator.cpp	2008-03-09 02:36:08.229521970 +0100
@@ -140,6 +140,29 @@
            urlStr = url.path();
        }
     }
+    else if (url.protocol() == "p7zip") {
+       bool stillInside = false;
+       if (KMimeType::findByPath(url.url(-1))
+           ->is("application/x-7z")) {
+           stillInside = true;
+       }
+       else {
+           KURL url1 = url.upURL();
+           while (url1 != url1.upURL()) {
+               if (KMimeType::findByPath(url1.url(-1))
+                   ->is("application/x-7z")) {
+                   stillInside = true;
+                   break;
+               }
+               url1 = url1.upURL();
+           }
+       }
+       if (!stillInside)
+       {
+           // Drop the zip:/ protocol since we are not in the zip anymore
+           urlStr = url.path();
+       }
+    }
     else if (url.protocol() == "tar")
     {
        bool stillInside = false;
