diff --git a/bass/Makefile b/bass/Makefile
index 90b91c2..5c9f607 100755
--- a/bass/Makefile
+++ b/bass/Makefile
@@ -1,8 +1,8 @@
 include ../nall/Makefile
 
 application := bass
-flags := $(flags) -O3 -fomit-frame-pointer -I..
-link := $(link) -s
+flags := $(flags) $(CXXFLAGS) -O3 -std=c++11 -fomit-frame-pointer -I..
+link := $(link)
 #flags := $(flags) -g -I..
 #link := $(link)
 objects := obj/bass.o
@@ -14,11 +14,7 @@ obj/bass.o: bass.cpp bass.hpp $(call rwildcard,core/*) $(call rwildcard,arch/*)
 	$(compiler) $(cppflags) $(flags) -o obj/bass.o -c bass.cpp
 
 install:
-ifneq ($(shell id -un),root)
-	$(error "make install must be run as root")
-else
-	cp $(application) /usr/local/bin/$(application)
-endif
+	install -D -m 755 $(application) $(DESTDIR)/usr/bin/$(application)
 
 uninstall:
 ifneq ($(shell id -un),root)
