Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(wersja 1030)
+++ CMakeLists.txt	(kopia robocza)
@@ -35,13 +35,7 @@
 
 # Set the version number with -DVERSION=X.X.X-uber
 IF( NOT VERSION )
-    FIND_PACKAGE(Subversion)
-    IF(SUBVERSION_FOUND)
-       Subversion_WC_INFO(${CMAKE_CURRENT_SOURCE_DIR} SVN_INFO)
-        SET( VERSION "1.8.0-SVN${SVN_INFO_WC_REVISION}" )
-    ELSE(SUBVERSION_FOUND)
-        SET( VERSION "1.8.0-SVN" )
-    ENDIF(SUBVERSION_FOUND)
+	SET( VERSION "1.8.0-SVN" )
 ENDIF( NOT VERSION )
 
 # Fill in SDLMAIN_LIBRARY on OS X manually to avoid using SDLMain.m
