diff -Nur fceu.orig/cmake/fceux.cmake fceu/cmake/fceux.cmake
--- fceu.orig/cmake/fceux.cmake	2008-06-18 01:40:08.000000000 +0200
+++ fceu/cmake/fceux.cmake	2009-01-14 20:54:45.114788597 +0100
@@ -193,6 +193,7 @@
   ${CMAKE_SOURCE_DIR}/src/utils/memory.cpp
   ${CMAKE_SOURCE_DIR}/src/utils/unzip.cpp
   ${CMAKE_SOURCE_DIR}/src/utils/xstring.cpp
+  ${CMAKE_SOURCE_DIR}/src/utils/ConvertUTF.c
 )
 
 set(SRC_DRIVERS_COMMON
