diff -Nur fceu.orig/CMakeLists.txt fceu/CMakeLists.txt
--- fceu.orig/CMakeLists.txt	2008-06-08 08:34:13.000000000 +0200
+++ fceu/CMakeLists.txt	2009-01-14 20:50:08.906788553 +0100
@@ -2,9 +2,6 @@
 project(fceux)
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_SOURCE_DIR}/bin)
 add_subdirectory(cmake/native)
-if(UNIX)
-  add_subdirectory(cmake/cross-mingw32)
-endif(UNIX)
 
 add_custom_command(
   OUTPUT ${EXECUTABLE_OUTPUT_PATH}/fceux.chm
