--- 3party/libjingle/libjingle.old	2006-08-10 12:09:44.000000000 +0200
+++ 3party/libjingle/libjingle.pro	2008-03-15 14:38:23.147047464 +0100
@@ -3,7 +3,7 @@
 TARGET = jingle
 
 win32:CONFIG += release
-unix:CONFIG += debug
+#unix:CONFIG += debug
 CONFIG += warn_off
 
 unix:debug: QMAKE_CXXFLAGS += -O0 -g3
--- src/src.old	2008-03-15 14:42:04.823047948 +0100
+++ src/src.pro	2008-03-15 14:42:09.855048160 +0100
@@ -11,7 +11,7 @@
 #DEFINES += GPG_DEBUG
 #DEFINES += XMPP_DEBUG
 
-CONFIG += jingle debug
+CONFIG += jingle
 INCLUDEPATH += ../3party/libjingle
 LIBS += -L../3party/libjingle -ljingle
 
--- 3party/jrtplib/jrtplib.old	2008-03-15 14:43:02.015047591 +0100
+++ 3party/jrtplib/jrtplib.pro	2008-03-15 14:43:08.295047745 +0100
@@ -1,5 +1,5 @@
 TEMPLATE = lib
-CONFIG += staticlib debug
+CONFIG += staticlib
 TARGET = jrtp
 INCLUDEPATH += .
 
