diff --git a/globalhotkeys.cpp b/globalhotkeys.cpp
index a53fbc7..076e0f9 100644
--- a/modules/globalhotkeys/globalhotkeys.cpp
+++ b/modules/globalhotkeys/globalhotkeys.cpp
@@ -1330,21 +1330,15 @@ void GlobalHotkeys::checkPendingHotkeys()
 			}
 			else if( hotkey->equals( hotkeys["TurnSilentModeOn"] ) )
 			{
-				notification_manager->silentActionActivated( NULL, true );
+				kadu->silentActionActivated( NULL, true );
 			}
 			else if( hotkey->equals( hotkeys["TurnSilentModeOff"] ) )
 			{
-				notification_manager->silentActionActivated( NULL, false );
+				kadu->silentActionActivated( NULL, false );
 			}
 			else if( hotkey->equals( hotkeys["ToggleSilentMode"] ) )
 			{
-				notification_manager->setSilentActionState();
-				bool silentmode = notification_manager->silent_mode;
-				if( notification_manager->silent_action->actions().count() > 0 )
-				{
-					silentmode = notification_manager->silent_action->actions().at(0)->isChecked();
-				}
-				notification_manager->silentActionActivated( NULL, !silentmode );
+				kadu->silentActionActivated( NULL, !kadu->silentMode() );
 			}
 			// <---
 			else
