--- Makefile.orig	2009-05-10 02:59:55.000000000 +0200
+++ Makefile	2009-05-12 17:47:28.101316902 +0200
@@ -7,7 +7,7 @@
 
 c     := $(compiler)
 cpp   := $(subst cc,++,$(compiler))
-flags := -O3 -fomit-frame-pointer -Ilib
+flags := $(CXXFLAGS) -fomit-frame-pointer -Ilib
 link  := -s
 
 # profile-guided instrumentation:
@@ -15,7 +15,7 @@
 # link += -lgcov
 
 # profile-guided optimization:
-flags += -fprofile-use
+#flags += -fprofile-use
 
 ################
 ### platform ###
