diff --git a/bsnes/ui/utility/utility.cpp b/bsnes/ui/utility/utility.cpp
index 9a4831a..25e7fa5 100755
--- a/bsnes/ui/utility/utility.cpp
+++ b/bsnes/ui/utility/utility.cpp
@@ -8,7 +8,6 @@ void Utility::setMode(Interface::Mode mode) {
   mainWindow->nesMenu.setVisible(false);
   mainWindow->snesMenu.setVisible(false);
   mainWindow->gameBoyMenu.setVisible(false);
-  mainWindow->viewport.setVisible(mode != Interface::Mode::None);
 
   if(mode == Interface::Mode::None) {
     mainWindow->setTitle(application->title);
