diff --git a/Makefile b/Makefile
index feb1171..ed9dca8 100755
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ ui = ui_qt
 
 c     := $(compiler)
 cpp   := $(subst cc,++,$(compiler))
-flags := -O3 -fomit-frame-pointer -Ilib
+flags := $(CXXFLAGS) -fomit-frame-pointer -Ilib
 link  :=
 
 # profile-guided instrumentation:
@@ -23,8 +23,6 @@ link  :=
 ################
 
 ifeq ($(platform),x)
-  link += -s
-
   ruby := video.glx video.xv video.qtraster video.sdl
   ruby += audio.alsa audio.openal audio.oss audio.pulseaudio audio.pulseaudiosimple audio.ao
   ruby += input.sdl input.x
