diff --git a/Makefile b/Makefile
index 9062cd3..b0ef8fd 100755
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ ui := ui_qt
 # compiler
 c       := $(compiler) --std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects :=
 
@@ -13,11 +13,11 @@ objects :=
 # link += -lgcov
 
 # profile-guided optimization
-flags += -fprofile-use
+# flags += -fprofile-use
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows -mthreads
@@ -60,8 +60,8 @@ endif
 install:
 ifeq ($(platform),x)
 	install -D -m 755 ../bsnes $(DESTDIR)$(prefix)/bin/bsnes
-	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
+	install -D -m 644 data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 endif
 
 uninstall:
