diff --git a/src/snes/ppu/bppu/mmio/mmio.cpp b/src/snes/ppu/bppu/mmio/mmio.cpp
index 29045ea..db592ba 100755
--- a/src/snes/ppu/bppu/mmio/mmio.cpp
+++ b/src/snes/ppu/bppu/mmio/mmio.cpp
@@ -571,7 +571,7 @@ uint8 r = 0x00;
 }
 
 uint8 bPPU::mmio_read(unsigned addr) {
-  synchronize_cpu();
+  cpu.synchronize_ppu();
 
   switch(addr & 0xffff) {
     case 0x2104:
@@ -610,7 +610,7 @@ uint8 bPPU::mmio_read(unsigned addr) {
 }
 
 void bPPU::mmio_write(unsigned addr, uint8 data) {
-  synchronize_cpu();
+  cpu.synchronize_ppu();
 
   switch(addr & 0xffff) {
     case 0x2100: return mmio_w2100(data);  //INIDISP
