diff --git a/src/Makefile b/src/Makefile
index c3c1fe9..4b5de6d 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -4,7 +4,7 @@ ui := ui_qt
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects :=
 
@@ -17,7 +17,7 @@ objects :=
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows -mthreads
@@ -60,8 +60,8 @@ endif
 install:
 ifeq ($(platform),x)
 	install -D -m 755 ../bsnes $(DESTDIR)$(prefix)/bin/bsnes
-	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
+	install -D -m 644 data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 endif
 
 uninstall:
