diff --git a/bsnes/Makefile b/bsnes/Makefile
index ce7eab5..e0dc207 100755
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ ui := ui-qt
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I. -I$(snes)
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I. -I$(snes)
 link    :=
 objects :=
 
@@ -19,7 +19,7 @@ objects :=
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows
@@ -56,15 +56,14 @@ ifeq ($(platform),osx)
 	test -d ../bsnes.app || mkdir -p ../bsnes.app/Contents/MacOS
 	$(strip $(cpp) -o ../bsnes.app/Contents/MacOS/bsnes $(objects) $(link))
 else
-	$(strip $(cpp) -o out/bsnes $(objects) $(link))
+	$(strip $(cpp) -o out/bsnes-$(profile) $(objects) $(link))
 endif
 
 install:
 ifeq ($(platform),x)
-	install -D -m 755 out/bsnes $(DESTDIR)$(prefix)/bin/bsnes
-	install -D -m 644 ui-qt/data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 ui-qt/data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
-	gconftool-2 --type bool --set /desktop/gnome/interface/menus_have_icons true
+	install -D -m 755 out/bsnes-$(profile) $(DESTDIR)$(prefix)/bin/bsnes-$(profile)
+	install -D -m 644 ui-qt/data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 ui-qt/data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 endif
 
 uninstall:
diff --git a/bsnes/launcher/cc.sh b/bsnes/launcher/cc.sh
index 964034d..c9a92d8 100755
--- a/launcher/cc.sh
+++ b/launcher/cc.sh
@@ -1,2 +1 @@
-clear
-g++ -std=gnu++0x -s -O3 -fomit-frame-pointer -I.. -o ../out/bsnes launcher.cpp
+g++ -std=gnu++0x ${CXXFLAGS} -fomit-frame-pointer -I.. -o ../out/bsnes launcher.cpp
diff --git a/snesfilter/Makefile b/snesfilter/Makefile
index 089b86e..08ba81b 100755
--- a/../snesfilter/Makefile
+++ b/../snesfilter/Makefile
@@ -5,12 +5,12 @@ include nall/qt/Makefile
 
 c     := $(compiler) -std=gnu99
 cpp   := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags := -O3 -I. -Iobj -fomit-frame-pointer $(qtinc)
+flags := $(CXXFLAGS) -I. -Iobj -fomit-frame-pointer $(qtinc)
 link  :=
 
 ifeq ($(platform),x)
   flags := -fPIC -fopenmp $(flags)
-  link += -s -fopenmp -lpthread -lgomp
+  link += -fopenmp -lpthread -lgomp
 else ifeq ($(platform),osx)
   flags := -fPIC -fopenmp $(flags)
   link += -fopenmp -lpthread -lgomp
@@ -73,9 +73,8 @@ endif
 
 install:
 ifeq ($(platform),x)
-	install -D -m 755 libsnesfilter.a $(DESTDIR)$(prefix)/lib
-	install -D -m 755 libsnesfilter.so $(DESTDIR)$(prefix)/lib
-	ldconfig -n $(DESTDIR)$(prefix)/lib
+	install -D -m 755 libsnesfilter.a $(DESTDIR)$(prefix)/lib/libsnesfilter.a
+	install -D -m 755 libsnesfilter.so $(DESTDIR)$(prefix)/lib/libsnesfilter.so
 else ifeq ($(platform),osx)
 	cp libsnesfilter.dylib /usr/local/lib/libsnesfilter.dylib
 endif
diff --git a/snesreader/Makefile b/snesreader/Makefile
index 761317b..868a0cc 100755
--- a/../snesreader/Makefile
+++ b/../snesreader/Makefile
@@ -5,12 +5,11 @@ include nall/qt/Makefile
 
 c     := $(compiler) -std=gnu99
 cpp   := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags := -O3 -I. -Iobj -fomit-frame-pointer
+flags := $(CXXFLAGS) -I. -Iobj -fomit-frame-pointer
 link  :=
 
 ifeq ($(platform),x)
   flags := -fPIC $(flags)
-  link += -s
 else ifeq ($(platform),osx)
   flags := -fPIC $(flags)
 endif
@@ -171,9 +170,8 @@ endif
 
 install:
 ifeq ($(platform),x)
-	install -D -m 755 libsnesreader.a $(DESTDIR)$(prefix)/lib
-	install -D -m 755 libsnesreader.so $(DESTDIR)$(prefix)/lib
-	ldconfig -n $(DESTDIR)$(prefix)/lib
+	install -D -m 755 libsnesreader.a $(DESTDIR)$(prefix)/lib/libsnesreader.a
+	install -D -m 755 libsnesreader.so $(DESTDIR)$(prefix)/lib/libsnesreader.so
 else ifeq ($(platform),osx)
 	cp libsnesreader.dylib /usr/local/lib/libsnesreader.dylib
 endif
diff --git a/supergameboy/Makefile b/supergameboy/Makefile
index 60409af..cbef9ac 100755
--- a/../supergameboy/Makefile
+++ b/../supergameboy/Makefile
@@ -2,14 +2,13 @@ include nall/Makefile
 
 c     := $(compiler) -std=gnu99
 cpp   := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags := -O3 -fomit-frame-pointer -I. -Icommon -Ilibgambatte/include -Ilibgambatte/src
+flags := $(CXXFLAGS) -fomit-frame-pointer -I. -Icommon -Ilibgambatte/include -Ilibgambatte/src
 link  :=
 
 ifeq ($(platform),osx)
   flags := -fPIC $(flags)
 else ifeq ($(platform),x)
   flags := -fPIC $(flags)
-  link += -s
 endif
 
 objects := supergameboy
@@ -113,9 +112,8 @@ install:
 ifeq ($(platform),osx)
 	cp libsupergameboy.dylib /usr/local/lib/libsupergameboy.dylib
 else
-	install -D -m 755 libsupergameboy.a $(DESTDIR)$(prefix)/lib
-	install -D -m 755 libsupergameboy.so $(DESTDIR)$(prefix)/lib
-	ldconfig -n $(DESTDIR)$(prefix)/lib
+	install -D -m 755 libsupergameboy.a $(DESTDIR)$(prefix)/lib/libsupergameboy.a
+	install -D -m 755 libsupergameboy.so $(DESTDIR)$(prefix)/lib/libsupergameboy.so
 endif
 
 clean:
