diff --git a/src/ui_qt/debugger/tools/memory.cpp b/src/ui_qt/debugger/tools/memory.cpp
index f9db487..f29f735 100755
--- a/src/ui_qt/debugger/tools/memory.cpp
+++ b/src/ui_qt/debugger/tools/memory.cpp
@@ -13,8 +13,8 @@ MemoryEditor::MemoryEditor() {
   setLayout(layout);
 
   editor = new HexEditor;
-  editor->reader = bind(&MemoryEditor::reader, this);
-  editor->writer = bind(&MemoryEditor::writer, this);
+  editor->reader = {&MemoryEditor::reader, this};
+  editor->writer = {&MemoryEditor::writer, this};
   editor->setFont(QFont(Style::Monospace));
   editor->setMinimumWidth((editor->lineWidth() + 3) * editor->fontMetrics().width(' '));
   editor->setMinimumHeight((16 + 1) * editor->fontMetrics().height());
diff --git a/src/ui_qt/debugger/tracer.cpp b/src/ui_qt/debugger/tracer.cpp
index c0ca2e8..0b99b16 100755
--- a/src/ui_qt/debugger/tracer.cpp
+++ b/src/ui_qt/debugger/tracer.cpp
@@ -63,8 +63,8 @@ Tracer::Tracer() {
   traceMaskCPU = new uint8_t[(1 << 24) >> 3]();
   traceMaskSMP = new uint8_t[(1 << 16) >> 3]();
 
-  SNES::cpu.step_event = bind(&Tracer::stepCpu, this);
-  SNES::smp.step_event = bind(&Tracer::stepSmp, this);
+  SNES::cpu.step_event = {&Tracer::stepCpu, this};
+  SNES::smp.step_event = {&Tracer::stepSmp, this};
 }
 
 Tracer::~Tracer() {
