diff --git a/bsnes/ui/tools/cheat-database.cpp b/bsnes/ui/tools/cheat-database.cpp
index a09ddbd..27fcb22 100755
--- a/bsnes/ui/tools/cheat-database.cpp
+++ b/bsnes/ui/tools/cheat-database.cpp
@@ -31,7 +31,8 @@ void CheatDatabase::create() {
 
 void CheatDatabase::findCodes() {
   string data;
-  data.readfile(path.home("cheats.xml"));
+  if(data.readfile(path.home("cheats.xml")) == false)
+    data.readfile("%GAMES_DATADIR%/bsnes/cheats.xml");
   if(auto position = strpos(data, SNES::cartridge.sha256())) {
     auto startPosition = strpos((const char*)data + position(), ">");
     auto endPosition = strpos((const char*)data + position(), "</cartridge>");
diff --git a/bsnes/ui/general/main-window.cpp b/bsnes/ui/general/main-window.cpp
index e0e8850..88fcdc6 100755
--- a/bsnes/ui/general/main-window.cpp
+++ b/bsnes/ui/general/main-window.cpp
@@ -402,6 +402,11 @@ void MainWindow::setupFiltersAndShaders() {
     #endif
     files = directory::files(folderPath, "*.filter");
   }
+  if(files.size() == 0)
+  {
+    folderPath = "%GAMES_DATADIR%/bsnes/filters/";
+    files = directory::files(folderPath, "*.filter");
+  }
   foreach(filename, files) {
     settingsVideoFilterName.append({ folderPath, filename });
   }
@@ -448,6 +453,11 @@ void MainWindow::setupFiltersAndShaders() {
     #endif
     files = directory::files(folderPath, { "*.", config.video.driver, ".shader" });
   }
+  if(files.size() == 0)
+  {
+    folderPath = "%GAMES_DATADIR%/bsnes/shaders/";
+    files = directory::files(folderPath, { "*.", config.video.driver, ".shader" });
+  }
   foreach(filename, files) {
     settingsVideoShaderName.append({ folderPath, filename });
   }
