diff --git a/Makefile b/Makefile
index 81f9656..2475b4e 100644
--- a/Makefile
+++ b/Makefile
@@ -36,9 +36,9 @@ ifneq ($(UNAME), Linux)
 	CPPFLAGS += -DBSD
 endif
 
-PREFIX = /usr/local
-BINDIR = $(PREFIX)/bin
-DATADIR = $(PREFIX)/share/nestopia
+PREFIX = %GAMES_PREFIX%
+BINDIR = %GAMES_BINDIR%
+DATADIR = %GAMES_DATADIR%/nestopia
 
 # OpenGL Support
 CPPFLAGS += -DINCLUDE_OPENGL
@@ -190,15 +190,16 @@ $(BIN): $(OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $(BIN) $^ $(LIBS)
 
 install:
-	mkdir -p $(DATADIR)/icons
-	mkdir -p $(PREFIX)/share/pixmaps
-	install -m 0755 $(BIN) $(BINDIR)
-	install -m 0644 source/unix/icons/nestopia.desktop $(DATADIR)
-	install -m 0644 NstDatabase.xml $(DATADIR)
-	install -m 0644 source/unix/icons/*.png $(DATADIR)/icons
-	install -m 0644 source/unix/icons/*.svg $(DATADIR)/icons
-	install -m 0644 source/unix/icons/nestopia.svg $(PREFIX)/share/pixmaps
-	xdg-desktop-menu install --novendor $(DATADIR)/nestopia.desktop
+	mkdir -p $(DESTDIR)$(DATADIR)/icons
+	mkdir -p $(DESTDIR)$(BINDIR)
+	mkdir -p $(DESTDIR)/usr/share/applications
+	mkdir -p $(DESTDIR)/usr/share/pixmaps
+	install -m 0755 $(BIN) $(DESTDIR)$(BINDIR)
+	install -m 0644 source/unix/icons/nestopia.desktop $(DESTDIR)/usr/share/applications
+	install -m 0644 NstDatabase.xml $(DESTDIR)$(DATADIR)
+	install -m 0644 source/unix/icons/*.png $(DESTDIR)$(DATADIR)/icons
+	install -m 0644 source/unix/icons/*.svg $(DESTDIR)$(DATADIR)/icons
+	install -m 0644 source/unix/icons/nestopia.svg $(DESTDIR)/usr/share/pixmaps
 
 uninstall:
 	xdg-desktop-menu uninstall $(DATADIR)/nestopia.desktop
