diff --git a/plugins/jabber_protocol/CMakeLists.txt b/plugins/jabber_protocol/CMakeLists.txt
index c796235..ee6f7b7 100644
--- a/plugins/jabber_protocol/CMakeLists.txt
+++ b/plugins/jabber_protocol/CMakeLists.txt
@@ -10,7 +10,7 @@ find_package (Kadu REQUIRED CONFIG)
 find_package (PkgConfig)
 pkg_search_module (IDN REQUIRED libidn)
 pkg_search_module (ZLIB REQUIRED zlib)
-pkg_search_module (QXMPP REQUIRED qxmpp)
+pkg_search_module (QXMPP REQUIRED qxmpp-qt5)
 
 if (NOT ZLIB_FOUND)
 	find_package (ZLIB REQUIRED)
