diff --git a/cc-gtk.sh b/cc-gtk.sh
index 30b4e71..76dd7b8 100644
--- a/cc-gtk.sh
+++ b/cc-gtk.sh
@@ -1,4 +1,4 @@
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c phoenix/phoenix.cpp `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c snespurify.cpp -DPHOENIX_GTK
-g++-4.5 -s -o snespurify-gtk snespurify.o phoenix.o `pkg-config --libs gtk+-2.0` -lX11
+g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer ${CXXFLAGS} -c phoenix/phoenix.cpp `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
+g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer ${CXXFLAGS} -c snespurify.cpp -DPHOENIX_GTK
+g++ -o snespurify-gtk snespurify.o phoenix.o `pkg-config --libs gtk+-2.0` -lX11
 rm *.o
diff --git a/cc-qt.sh b/cc-qt.sh
index 6a063a4..3946c9b 100644
--- a/cc-qt.sh
+++ b/cc-qt.sh
@@ -1,5 +1,5 @@
 moc -i -o phoenix/qt/qt.moc phoenix/qt/qt.moc.hpp
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c phoenix/phoenix.cpp `pkg-config --cflags QtCore QtGui` -DPHOENIX_QT
-g++-4.5 -std=gnu++0x -I. -O3 -fomit-frame-pointer -c snespurify.cpp -DPHOENIX_QT
-g++-4.5 -s -o snespurify-qt snespurify.o phoenix.o `pkg-config --libs QtCore QtGui`
+g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer ${CXXFLAGS} -c phoenix/phoenix.cpp `pkg-config --cflags QtCore QtGui` -DPHOENIX_QT
+g++ -std=gnu++0x -I. -O3 -fomit-frame-pointer ${CXXFLAGS} -c snespurify.cpp -DPHOENIX_QT
+g++ -o snespurify-qt snespurify.o phoenix.o `pkg-config --libs QtCore QtGui`
 rm *.o
