diff -Nur supertux-0.3.1.orig/src/random_generator.cpp supertux-0.3.1/src/random_generator.cpp
--- supertux-0.3.1.orig/src/random_generator.cpp	2008-01-03 21:59:27.000000000 +0100
+++ supertux-0.3.1/src/random_generator.cpp	2009-06-02 17:56:22.920497466 +0200
@@ -40,6 +40,7 @@
 #include <time.h>
 #include <cassert>
 #include "random_generator.hpp"
+#include <cstdio>
 
 RandomGenerator systemRandom;               // global random number generator
 
