--- CMakeLists.txt.old	2009-12-25 11:33:05.337735626 +0100
+++ CMakeLists.txt	2009-12-25 11:47:22.447297075 +0100
@@ -49,7 +49,7 @@
 
 ## For autopackage
 OPTION(ENABLE_BINRELOC "Enable autopackage's BINRELOC features" OFF)
-SET(APPDATADIR "${CMAKE_INSTALL_PREFIX}/share/games/supertux2")
+SET(APPDATADIR "${CMAKE_INSTALL_PREFIX}/supertux2")
 
 ## Check endianess
 
@@ -368,7 +368,7 @@
 
 ELSE(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
-  SET(INSTALL_SUBDIR_BIN "games" CACHE STRING "Installation subdir for binaries")
+  SET(INSTALL_SUBDIR_BIN "bin" CACHE STRING "Installation subdir for binaries")
   SET(INSTALL_SUBDIR_SHARE "share/games/supertux2" CACHE STRING "Installation subdir for data")
   SET(INSTALL_SUBDIR_DOC "share/doc/supertux2" CACHE STRING "Installation subdir for docs")
 
