--- files/tar.1	2007-12-10 13:55:51.000000000 +0100
+++ /tmp/tar.1-modified	2007-12-10 14:07:14.000000000 +0100
@@ -20,6 +20,7 @@
 .I Common Options:
 .nf
 .BR -C ", " --directory " DIR"
+.BR -e ", " --7zg
 .BR -f ", " --file " F"
 .BR -j ", " --bzip2
 .BR -p ", " --preserve-permissions
@@ -51,6 +52,9 @@
 .BR --checkpoint
 ]
 [
+.BR -e ", " --7zg
+]
+[
 .BR -f ", " --file " [HOSTNAME:]F"
 ]
 [
@@ -304,6 +308,9 @@
 .B tar -xjf foo.tar.bz2 -C bar/
 extract bzipped foo.tar.bz2 after changing directory to bar
 .TP
+.B tar -cef foo.tar.7z bar/
+create 7zipped tar archive of the directory bar called foo.tar.7z
+.TP
 .B tar -xzf foo.tar.gz blah.txt
 extract the file blah.txt from foo.tar.gz
 .SH "FUNCTION LETTERS"
@@ -338,6 +345,9 @@
 .BR -C ", " --directory " DIR"
 change to directory DIR
 .TP
+.BR -e ", " --7zg
+filter archive through 7zip, use to decompress .7z files
+.TP
 .BR -f ", " --file " [HOSTNAME:]F"
 use archive file or device F (default "-", meaning stdin/stdout)
 .TP
@@ -376,6 +386,9 @@
 .BR --checkpoint
 print directory names while reading the archive
 .TP
+.BR -e ", " --7zg
+filter archive through 7zip, use to decompress .7z files
+.TP
 .BR -f ", " --file " [HOSTNAME:]F"
 use archive file or device F (default "-", meaning stdin/stdout)
 .TP
