diff --git a/Quake/Makefile b/Quake/Makefile
index 6c10d3d..7f17557 100644
--- a/Quake/Makefile
+++ b/Quake/Makefile
@@ -10,9 +10,9 @@
 USE_QS_CONBACK=1
 
 ### Enable/Disable codecs for streaming music support
-USE_CODEC_WAVE=1
-USE_CODEC_MP3=1
-USE_CODEC_VORBIS=1
+USE_CODEC_WAVE=0
+USE_CODEC_MP3=0
+USE_CODEC_VORBIS=0
 
 # which library to use for mp3 decoding: mad or mpg123
 MP3LIB=mad
@@ -59,7 +59,7 @@ endif
 
 CC ?= gcc
 WINDRES ?= windres
-STRIP ?= strip
+STRIP ?= echo
 
 #CPUFLAGS= -mtune=i686
 #CPUFLAGS= -march=pentium4
@@ -272,7 +272,7 @@ OBJS := strlcat.o \
 # ------------------------
 
 quakespasm:	$(OBJS)
-	$(CC) $(OBJS) $(X11_LFLAGS) $(LIBS) $(SDL_LIBS) -o $@
+	$(CC) $(OBJS) $(X11_LFLAGS) $(LDFLAGS) $(LIBS) $(SDL_LIBS) -o $@
 	$(call do_strip,$@)
 
 release:	quakespasm
