diff --git a/Quake/sys_sdl.c b/Quake/sys_sdl.c
index b240c5a..376bcfe 100644
--- a/Quake/sys_sdl.c
+++ b/Quake/sys_sdl.c
@@ -20,9 +20,17 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 
 */
 
-#include "quakedef.h"
+#include <sys/types.h>
+#ifdef _WIN32
+#include <io.h>
+#include <direct.h>
+#else
+#include <sys/stat.h>
+#endif
 #include "errno.h"
 
+#include "quakedef.h"
+
 #define CONSOLE_ERROR_TIMEOUT	60.0	/* # of seconds to wait on Sys_Error running */
 qboolean		isDedicated;
 static qboolean		sc_return_on_enter = false;
@@ -42,7 +50,7 @@ int findhandle (void)
 	return -1;
 }
 
-int filelength (FILE *f)
+int Sys_filelength (FILE *f)
 {
 	int pos;
 	int end;
@@ -72,7 +80,7 @@ int Sys_FileOpenRead (char *path, int *hndl)
 	{
 		sys_handles[i] = f;
 		*hndl = i;
-		retval = filelength(f);
+		retval = Sys_filelength(f);
 	}
 
 	return retval;
@@ -131,6 +139,13 @@ int Sys_FileTime (char *path)
 
 void Sys_mkdir (char *path)
 {
+#ifdef _WIN32
+	int rc = _mkdir (path);
+#else
+	int rc = mkdir (path, 0777);
+#endif
+	if (rc != 0 && errno != EEXIST)
+		Sys_Error("Unable to create directory %s", path);
 }
 
 
