diff --git a/snesreader/Makefile b/snesreader/Makefile
index 761317b..868a0cc 100755
--- a/snesreader/Makefile
+++ b/snesreader/Makefile
@@ -5,12 +5,11 @@ include nall/qt/Makefile
 
 c     := $(compiler) -std=gnu99
 cpp   := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags := -O3 -I. -Iobj -fomit-frame-pointer
+flags := $(CXXFLAGS) -I. -Iobj -fomit-frame-pointer
 link  :=
 
 ifeq ($(platform),x)
   flags := -fPIC $(flags)
-  link += -s
 else ifeq ($(platform),osx)
   flags := -fPIC $(flags)
 endif
