--- branches/0.6.5/modules/weather/weather_cfg_ui_handler.cpp	2009/05/17 21:08:25	6654
+++ branches/0.6.5/modules/weather/weather_cfg_ui_handler.cpp	2009/05/18 14:46:59	6657
@@ -91,6 +91,8 @@
 
 void WeatherCfgUiHandler::configurationUpdated()
 {
+	if (!MainConfigurationWindow::instance()->isShown()) return;
+
 	for( int i = 0; i < serverList_->topLevelItemCount(); ++i )
 	{
 		QTreeWidgetItem* item = serverList_->topLevelItem( i );
