diff -Nur qsopcast-0.3.5.orig/src/main.cpp qsopcast-0.3.5/src/main.cpp
--- qsopcast-0.3.5.orig/src/main.cpp	2007-03-27 07:47:57.000000000 +0200
+++ qsopcast-0.3.5/src/main.cpp	2009-02-11 16:05:43.898874571 +0100
@@ -21,6 +21,7 @@
 #include <qtextcodec.h>
 #include "mainwindow.h"
 #include "channel.h"
+#include <cstdlib>
 
 int
 main (int argc, char **argv)
diff -Nur qsopcast-0.3.5.orig/src/record.cpp qsopcast-0.3.5/src/record.cpp
--- qsopcast-0.3.5.orig/src/record.cpp	2007-04-25 07:45:47.000000000 +0200
+++ qsopcast-0.3.5/src/record.cpp	2009-02-12 20:04:01.000000000 +0100
@@ -29,7 +29,7 @@
 #include "menubar.h"
 #include "utils.h"
 #include "mystatusbar.h"
-
+#include <cstdlib>
 
 Record::Record (QWidget * parent, const char *name):
 MyHBox (parent, name),
diff -Nur qsopcast-0.3.5.orig/src/sopfork.cpp qsopcast-0.3.5/src/sopfork.cpp
--- qsopcast-0.3.5.orig/src/sopfork.cpp	2007-05-25 16:43:24.000000000 +0200
+++ qsopcast-0.3.5/src/sopfork.cpp	2009-02-12 20:09:29.000000000 +0100
@@ -27,6 +27,7 @@
 #include "mystatusbar.h"
 #include "record.h"
 #include "mainwindow.h"
+#include <cstdlib>
 
 SopFork::SopFork (QWidget * widget, QString url, QString name, QString type,
 		  QWidget * parent, const char *name2):
