diff --git a/icons/babelfish.xpm b/icons/babelfish.xpm
index bf69eb0..62dcc82 100644
--- a/icons/babelfish.xpm
+++ b/icons/babelfish.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * babelfish_xpm[] = {
+static const char * babelfish_xpm[] = {
 "22 22 155 2",
 "  	c None",
 ". 	c #000000",
diff --git a/icons/babelfish_small.xpm b/icons/babelfish_small.xpm
index 6dddb1e..ce04883 100644
--- a/icons/babelfish_small.xpm
+++ b/icons/babelfish_small.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * babelfish_small_xpm[] = {
+static const char * babelfish_small_xpm[] = {
 "16 16 120 2",
 "  	c None",
 ". 	c #010103",
diff --git a/icons/clipboard.xpm b/icons/clipboard.xpm
index 62bfc7f..087fcb9 100644
--- a/icons/clipboard.xpm
+++ b/icons/clipboard.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * clipboard_xpm[] = {
+static const char * clipboard_xpm[] = {
 "22 22 145 2",
 "  	c None",
 ". 	c #792D00",
diff --git a/icons/conf.xpm b/icons/conf.xpm
index 72dec59..94e6853 100644
--- a/icons/conf.xpm
+++ b/icons/conf.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * conf_xpm[] = {
+static const char * conf_xpm[] = {
 "16 16 17 1",
 " 	c None",
 ".	c #89899D",
diff --git a/icons/configure.xpm b/icons/configure.xpm
index 420d635..f8895d3 100644
--- a/icons/configure.xpm
+++ b/icons/configure.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * configure_xpm[] = {
+static const char * configure_xpm[] = {
 "22 22 162 2",
 "  	c None",
 ". 	c #B8B8C7",
diff --git a/icons/de_pl.xpm b/icons/de_pl.xpm
index bfd0bcf..e3c206b 100644
--- a/icons/de_pl.xpm
+++ b/icons/de_pl.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * de_pl_xpm[] = {
+static const char * de_pl_xpm[] = {
 "22 22 5 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/de_pl_dis.xpm b/icons/de_pl_dis.xpm
index da1fe91..7c58df8 100644
--- a/icons/de_pl_dis.xpm
+++ b/icons/de_pl_dis.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * de_pl_dis_xpm[] = {
+static const char * de_pl_dis_xpm[] = {
 "22 22 4 1",
 " 	g None",
 ".	g #767676",
diff --git a/icons/en_pl.xpm b/icons/en_pl.xpm
index d16f337..93467c5 100644
--- a/icons/en_pl.xpm
+++ b/icons/en_pl.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * en_pl_xpm[] = {
+static const char * en_pl_xpm[] = {
 "22 22 5 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/en_pl_dis.xpm b/icons/en_pl_dis.xpm
index 75f3c51..95fc1cd 100644
--- a/icons/en_pl_dis.xpm
+++ b/icons/en_pl_dis.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * en_pl_dis_xpm[] = {
+static const char * en_pl_dis_xpm[] = {
 "22 22 5 1",
 " 	g None",
 ".	g #757575",
diff --git a/icons/f1.xpm b/icons/f1.xpm
index a030239..e7864db 100644
--- a/icons/f1.xpm
+++ b/icons/f1.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f1_xpm[] = {
+static const char * f1_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f10.xpm b/icons/f10.xpm
index 4c0435d..5f1ffea 100644
--- a/icons/f10.xpm
+++ b/icons/f10.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f10_xpm[] = {
+static const char * f10_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f11.xpm b/icons/f11.xpm
index ef9d9b4..021dd80 100644
--- a/icons/f11.xpm
+++ b/icons/f11.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f11_xpm[] = {
+static const char * f11_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f2.xpm b/icons/f2.xpm
index 62dadb4..0142821 100644
--- a/icons/f2.xpm
+++ b/icons/f2.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f2_xpm[] = {
+static const char * f2_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f3.xpm b/icons/f3.xpm
index 0be2d75..d9aae2c 100644
--- a/icons/f3.xpm
+++ b/icons/f3.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f3_xpm[] = {
+static const char * f3_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f4.xpm b/icons/f4.xpm
index 11b256e..4df9adc 100644
--- a/icons/f4.xpm
+++ b/icons/f4.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f4_xpm[] = {
+static const char * f4_xpm[] = {
 "5 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f5.xpm b/icons/f5.xpm
index c87f441..f000522 100644
--- a/icons/f5.xpm
+++ b/icons/f5.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f5_xpm[] = {
+static const char * f5_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f6.xpm b/icons/f6.xpm
index 79fb9bd..3daa192 100644
--- a/icons/f6.xpm
+++ b/icons/f6.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f6_xpm[] = {
+static const char * f6_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f7.xpm b/icons/f7.xpm
index 7d57121..5b606a6 100644
--- a/icons/f7.xpm
+++ b/icons/f7.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f7_xpm[] = {
+static const char * f7_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f8.xpm b/icons/f8.xpm
index 798533b..10886aa 100644
--- a/icons/f8.xpm
+++ b/icons/f8.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f8_xpm[] = {
+static const char * f8_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/f9.xpm b/icons/f9.xpm
index 51b24cd..345c0c7 100644
--- a/icons/f9.xpm
+++ b/icons/f9.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * f9_xpm[] = {
+static const char * f9_xpm[] = {
 "7 12 2 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/loop.xpm b/icons/loop.xpm
index dc0de35..a655b3e 100644
--- a/icons/loop.xpm
+++ b/icons/loop.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * loop_xpm[] = {
+static const char * loop_xpm[] = {
 "22 22 57 1",
 " 	c None",
 ".	c #024266",
diff --git a/icons/pl_de.xpm b/icons/pl_de.xpm
index 5e0d277..25b5c3f 100644
--- a/icons/pl_de.xpm
+++ b/icons/pl_de.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pl_de_xpm[] = {
+static const char * pl_de_xpm[] = {
 "22 22 5 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/pl_de_dis.xpm b/icons/pl_de_dis.xpm
index 5af09c1..7da38af 100644
--- a/icons/pl_de_dis.xpm
+++ b/icons/pl_de_dis.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pl_de_dis_xpm[] = {
+static const char * pl_de_dis_xpm[] = {
 "22 22 4 1",
 " 	g None",
 ".	g #767676",
diff --git a/icons/pl_en.xpm b/icons/pl_en.xpm
index 649a6a7..f790b1d 100644
--- a/icons/pl_en.xpm
+++ b/icons/pl_en.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pl_en_xpm[] = {
+static const char * pl_en_xpm[] = {
 "22 22 5 1",
 " 	c None",
 ".	c #000000",
diff --git a/icons/pl_en_dis.xpm b/icons/pl_en_dis.xpm
index 2258e7c..35dd0b9 100644
--- a/icons/pl_en_dis.xpm
+++ b/icons/pl_en_dis.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * pl_en_dis_xpm[] = {
+static const char * pl_en_dis_xpm[] = {
 "22 22 5 1",
 " 	g None",
 ".	g #767676",
diff --git a/icons/player_play.xpm b/icons/player_play.xpm
index 436c096..0bec592 100644
--- a/icons/player_play.xpm
+++ b/icons/player_play.xpm
@@ -1,5 +1,5 @@
 /* XPM */
-static char * player_play_xpm[] = {
+static const char * player_play_xpm[] = {
 "24 24 46 1",
 " 	c None",
 ".	c #6E6E6E",
diff --git a/src/engine_sap.cpp b/src/engine_sap.cpp
index d241780..612f279 100644
--- a/src/engine_sap.cpp
+++ b/src/engine_sap.cpp
@@ -164,7 +164,7 @@ int EngineSAP::ReadDefinition(int index)
     return 0;
 }
 
-struct okreslenia { char *nazwa; int flagi; int maska; } okreslenia[]={
+struct okreslenia { const char *nazwa; int flagi; int maska; } okreslenia[]={
 {"przymiotnik",1,15},
 {"przys³ówek",2,15},
 {"spójnik",3,15},
