diff --git a/bsnes/Makefile b/bsnes/Makefile
old mode 100644
new mode 100755
index d4ac996..849a77b
--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -20,11 +20,11 @@ objects :=
 ifeq ($(DEBUG), 1)
   flags += -O0 -g
 else
-  flags += -O3 -fomit-frame-pointer
+  flags += $(CXXFLAGS) -fomit-frame-pointer
   ifeq ($(platform),osx)
     # -s is deprecated as of clang 7.0, and using it breaks linking for some reason
   else
-    link += -s
+    #link += -s
   endif
 endif
 
@@ -89,11 +89,8 @@ endif
 install: build plugins
 ifeq ($(platform),x)
 	install -D -m 755 out/bsnes $(DESTDIR)$(prefix)/bin/bsnes
-	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
-	test -d ~/.bsnes || mkdir ~/.bsnes
-	cp data/cheats.xml ~/.bsnes/cheats.xml
-	chmod 777 ~/.bsnes ~/.bsnes/cheats.xml
+	install -D -m 644 data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 else ifeq ($(platform),osx)
 	mv -f $(osxbundle) /Applications/bsnes+.app
 	test -d ~/.bsnes || mkdir ~/.bsnes
