Index: include/osgART/Marker
===================================================================
--- include/osgART/Marker	(wersja 127)
+++ include/osgART/Marker	(kopia robocza)
@@ -39,6 +39,7 @@
 // STL headers
 #include <string>
 #include <vector>
+#include <typeinfo>
 
 // local headers
 #include "osgART/Export"
Index: src/osgART/GenericTracker.cpp
===================================================================
--- src/osgART/GenericTracker.cpp	(wersja 127)
+++ src/osgART/GenericTracker.cpp	(kopia robocza)
@@ -40,6 +40,7 @@
 #include <osg/Notify>
 
 #include <iostream>
+#include <cstring>
 
 
 namespace osgART {
Index: src/osgART/Video/ARToolKit/ARToolKitVideo
===================================================================
--- src/osgART/Video/ARToolKit/ARToolKitVideo	(wersja 129)
+++ src/osgART/Video/ARToolKit/ARToolKitVideo	(kopia robocza)
@@ -62,6 +62,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdlib>
 
 //graphics include
 #include <AR/video.h>
