--- Osmose-0-8-5b-source.orig/Makefile	2009-01-07 20:08:00.000000000 +0100
+++ Osmose-0-8-5b-source/Makefile	2009-01-16 17:47:41.387476397 +0100
@@ -42,7 +42,7 @@
 
 # Compiler/Linker Flags
 # add -O3 to CFLAGS for speedup -pg on LFLAGS and CFLAGS for gprof
-CFLAGS =  -Wall -D__USE_UNIX98 -O3
+CFLAGS +=  -Wall -D__USE_UNIX98
 LFLAGS =  -lSDL -lGL -lz
 AFLAGS =  -f elf
 CC  = g++
@@ -53,10 +53,10 @@
 .cpp.o:
 	$(CC) $(CFLAGS) -c $*.cpp -o $*.o
 
-all:  libs disasm osmose stripexe
+all:  libs disasm osmose
 
 osmose: $(z80_lib) $(OSM_OBJS) $(OPT_OBJS)
-	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) $(UNZIP_LIB) $(Z_LIB) -o $(OSM_EXE) $(LFLAGS)
+	+$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) $(UNZIP_LIB) $(Z_LIB) -o $(OSM_EXE) $(LFLAGS)
 
 disasm: $(DISASM_OBJS)
 
@@ -75,7 +75,7 @@
 	rm -f *.o core $(OSM_EXE)
 
 libs:
-	cd $(UNZIP_DIR) && make unzip.a
+	+cd $(UNZIP_DIR) && make unzip.a
 
 
 
