diff --git a/Makefile b/Makefile
index bf173a4..28dd53d 100644
--- a/Makefile
+++ b/Makefile
@@ -101,8 +101,9 @@ tools/ssnes-joyconfig: $(JOYCONFIG_OBJ)
 	@$(if $(Q), $(shell echo echo CC $<),)
 
 install: $(TARGET)
-	install -m755 $(TARGET) $(DESTDIR)$(PREFIX)/bin 
-	install -m644 ssnes.cfg $(DESTDIR)/etc/ssnes.cfg
+	install -dm755 $(DESTDIR)$(PREFIX)/bin $(DESTDIR)$(PREFIX)/share/man/man1
+	install -m755 $(TARGET) $(DESTDIR)$(PREFIX)/bin
+	install -D -m644 ssnes.cfg $(DESTDIR)/etc/ssnes.cfg
 	install -m644 docs/{ssnes,ssnes-joyconfig}.1 $(DESTDIR)$(PREFIX)/share/man/man1
 
 uninstall:
diff --git a/qb/qb.params.sh b/qb/qb.params.sh
index 231310d..f8cb541 100644
--- a/qb/qb.params.sh
+++ b/qb/qb.params.sh
@@ -136,8 +136,7 @@ parse_input()
             exit 0
             ;;
          *)
-            print_help
-            exit 1
+            echo "Unknown option: $1"
             ;;
 
       esac
