diff --git a/plugins/qt4_docking/CMakeLists.txt b/plugins/qt4_docking/CMakeLists.txt
index cc35dee..3330413 100644
--- a/plugins/qt4_docking/CMakeLists.txt
+++ b/plugins/qt4_docking/CMakeLists.txt
@@ -11,6 +11,10 @@ include (FindKadu)
 include_directories (${KADU_INCLUDE_DIR})
 include_directories (${KADU_INCLUDE_DIR}/kadu-core)
 
+kadu_api_directories (plugins/qt4_docking
+	.
+)
+
 set (SOURCES
 	qt4_docking.cpp
 	qt4-docking-plugin.cpp
