diff --git a/src/IOFunctions.cpp b/src/IOFunctions.cpp
index aa435b8..13cb328 100644
--- a/src/IOFunctions.cpp
+++ b/src/IOFunctions.cpp
@@ -332,10 +332,15 @@ NONS_RedirectedOutput::~NONS_RedirectedOutput(){
 }
 
 void NONS_RedirectedOutput::write_to_stream(const std::stringstream &str){
+#if NONS_SYS_WINDOWS
 	if (CLOptions.verbosity>=255 && this->vc)
 		this->vc->put(str.str());
 	else if (CLOptions.override_stdout && this->file)
 		(*this->file) <<str.str();
+#else
+	if (CLOptions.override_stdout && this->file)
+		(*this->file) <<str.str();
+#endif
 	else
 		this->cout <<str.str();
 }
