diff -Nur kadu.old/modules/amarok2_mediaplayer/CMakeLists.txt kadu/modules/amarok2_mediaplayer/CMakeLists.txt
--- kadu.old/modules/amarok2_mediaplayer/CMakeLists.txt	2009-05-15 16:40:51.000000000 +0200
+++ kadu/modules/amarok2_mediaplayer/CMakeLists.txt	2009-09-16 00:40:53.715013928 +0200
@@ -22,3 +22,5 @@
 else (amarok2_mediaplayer STREQUAL STATIC)
 	install (TARGETS amarok2_mediaplayer DESTINATION ${MODULEDIR})
 endif (amarok2_mediaplayer STREQUAL STATIC)
+
+target_link_libraries (amarok2_mediaplayer ${QT_QTDBUS_LIBRARY})
