--- configure.ac.orig	2008-11-12 12:05:05.000000000 +0100
+++ configure.ac	2008-11-12 12:13:06.224467294 +0100
@@ -22,13 +22,8 @@
     [Define to be the subversion revision number])
 fi
 
-CFLAGS=
-CPPFLAGS=
-CXXFLAGS=
-CCASFLAGS=
-
 WARNING_FLAGS="-Wall -Wno-format -Wno-unused-value"
-NORMAL_FLAGS=" -pipe -O3 -fomit-frame-pointer ${WARNING_FLAGS}"
+NORMAL_FLAGS=" -pipe -finline-functions -fomit-frame-pointer"
 DEBUG_FLAGS=" -g -O0 ${WARNING_FLAGS}"
 
 dnl Check for debug build
