diff --git a/Makefile b/Makefile
index 946491e..898824d 100755
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@ ui := ui
 # compiler
 c       := $(compiler) -std=gnu99
 cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
-flags   := -O3 -fomit-frame-pointer -I.
+flags   := $(CXXFLAGS) -fomit-frame-pointer -I.
 link    :=
 objects := libco
 
@@ -20,7 +20,7 @@ objects := libco
 
 # platform
 ifeq ($(platform),x)
-  link += -s -ldl -lX11 -lXext
+  link += -ldl -lX11 -lXext
 else ifeq ($(platform),osx)
 else ifeq ($(platform),win)
   link += -mwindows
@@ -66,11 +66,8 @@ ifeq ($(platform),x)
 	install -D -m 755 out/$(output) $(DESTDIR)$(prefix)/bin/$(output)
 endif
 ifeq ($(output),bsnes)
-	install -D -m 644 data/bsnes.png $(DESTDIR)$(prefix)/share/pixmaps/bsnes.png
-	install -D -m 644 data/bsnes.desktop $(DESTDIR)$(prefix)/share/applications/bsnes.desktop
-	test -d ~/.bsnes || mkdir ~/.bsnes
-	cp data/cheats.xml ~/.bsnes/cheats.xml
-	chmod 777 ~/.bsnes ~/.bsnes/cheats.xml
+	install -D -m 644 data/bsnes.png $(DESTDIR)/usr/share/pixmaps/bsnes.png
+	install -D -m 644 data/bsnes.desktop $(DESTDIR)/usr/share/applications/bsnes.desktop
 endif
 
 uninstall:
