--- configure.in.orig	2008-10-20 17:15:35.000000000 +0200
+++ configure.in	2008-10-20 18:09:38.016971339 +0200
@@ -334,19 +334,6 @@
 CPPFLAGS="$aegisub_save_CPPFLAGS"
 fi
 
-AC_AGI_COMPILE([PortAudio], [portaudio], [$CFLAGS $PTHREAD_CFLAGS], [$LDFLAGS $PORTAUDIO_LDFLAGS],[
-#include <portaudio.h>
-int main(void) {
-  PaError err = Pa_Initialize();
-  if (err != paNoError) return 1;
-  return 0;
-}])
-
-if test "$agi_with_portaudio" = "no" && test "$with_portaudio" = "yes"; then
-  AC_MSG_WARN([PortAudio detected, but it doesn't work...])
-  with_portaudio="no"
-fi
-
 if test "$with_portaudio" = "yes"; then
   AC_DEFINE(WITH_PORTAUDIO, 1, [Enable PortAudio Audio Provider])
 fi
