diff --git a/plugins/qt4_docking_notify/CMakeLists.txt b/plugins/qt4_docking_notify/CMakeLists.txt
index 8b0994b..77a2bd1 100644
--- a/plugins/qt4_docking_notify/CMakeLists.txt
+++ b/plugins/qt4_docking_notify/CMakeLists.txt
@@ -9,6 +9,7 @@ set (KADU_FIND_REQUIRED true)
 include (FindKadu)
 
 include_directories (${KADU_INCLUDE_DIR}/kadu-core)
+include_directories (${KADU_INCLUDE_DIR})
 
 set (SOURCES
 	qt4-docking-notify.cpp
