diff --git a/plugins/simpleview/CMakeLists.txt b/plugins/simpleview/CMakeLists.txt
index 639fe2f..64042c2 100644
--- a/plugins/simpleview/CMakeLists.txt
+++ b/plugins/simpleview/CMakeLists.txt
@@ -1,3 +1,16 @@
+project (simpleview)
+
+# minimal required versions
+cmake_minimum_required (VERSION 2.8)
+set (QT_MIN_VERSION "4.7.0")
+set (CMAKE_MIN_VERSION "2.6.0")
+
+set (KADU_FIND_REQUIRED true)
+include (FindKadu)
+
+include_directories (${KADU_INCLUDE_DIR})
+include_directories (${KADU_INCLUDE_DIR}/kadu-core)
+
 set (SOURCES
 	simpleview.cpp
 	simpleview-config-ui.cpp
