diff --git a/dist/clementine.desktop b/dist/clementine.desktop
index ac2671f..9eb0d53 100644
--- a/dist/clementine.desktop
+++ b/dist/clementine.desktop
@@ -6,7 +6,7 @@ GenericName=Clementine Music Player
 Comment=Plays music and last.fm streams
 Exec=clementine
 TryExec=clementine
-Icon=application-x-clementine
+Icon=clementine
 Terminal=false
 Categories=AudioVideo;Player;Qt;
 StartupNotify=false
