diff --git a/cmake/modules/FindFfmpeg.cmake b/cmake/modules/FindFfmpeg.cmake
index 2c44a92..7a100d6 100644
--- a/cmake/modules/FindFfmpeg.cmake
+++ b/cmake/modules/FindFfmpeg.cmake
@@ -62,6 +62,8 @@ FOREACH( INC_ ${FFMPEG_HEADERS} )
    SET( TMP_ TMP-NOTFOUND )
 ENDFOREACH( INC_ )
 
+string(REPLACE "/usr/include/" "" FFMPEG_INCLUDE_DIR "${FFMPEG_INCLUDE_DIR}")
+
 # clear out duplicates
 #LIST( REMOVE_DUPLICATES FFMPEG_INCLUDE_DIR )
 #LIST( REMOVE_DUPLICATES FFMPEG_LIBRARY_DIR )
diff --git a/lib/videofile.cpp b/lib/videofile.cpp
index e3822bc..6ed9ed0 100644
--- a/lib/videofile.cpp
+++ b/lib/videofile.cpp
@@ -58,7 +58,6 @@ bool VideoFile::open(const QString& filename)
     static bool init=false;
     if(!init) {
         av_register_all();
-        avcodec_init();
         avcodec_register_all();
         av_log_set_level(0);
         init=true;
@@ -66,7 +65,7 @@ bool VideoFile::open(const QString& filename)
 
     close();
 
-    if ( av_open_input_file(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0, 0L) != 0 ||
+    if ( avformat_open_input(&avFromatContext, QFile::encodeName(filename).constData(), 0L, 0L) != 0 ||
          av_find_stream_info(avFromatContext) < 0) {
         close();
     }
