diff --git a/unix/Makefile.in b/unix/Makefile.in
index a20788e..15a38d1 100644
--- a/unix/Makefile.in
+++ b/unix/Makefile.in
@@ -15,7 +15,7 @@ CFLAGS     = $(CCFLAGS)
 
 .SUFFIXES: .o .cpp .c .cc .h .m .i .s .obj
 
-all: Makefile configure libsnes.so
+all: Makefile configure libsnes.so libsnes.a
 
 Makefile: configure Makefile.in
 	@echo "Makefile is older than configure or in-file. Run configure or touch Makefile."
@@ -26,7 +26,10 @@ configure: configure.ac
 	exit 1
 
 libsnes.so: $(OBJECTS)
-	$(CCC) -fPIC -shared $(INCLUDES) -o $@ $(OBJECTS) -lm @S9XLIBS@
+	$(CCC) -fPIC -shared $(INCLUDES) -Wl,-soname,libsnes.so.1 -o $@ $(OBJECTS) -lm @S9XLIBS@
+
+libsnes.a: $(OBJECTS)
+	ar rcs $@ $(OBJECTS)
 
 .cpp.o:
 	$(CCC) $(INCLUDES) -c $(CCFLAGS) $*.cpp -o $@
