diff --git a/src/hw_cpu/v810/fpu-new/mednafen-gcc.h b/src/hw_cpu/v810/fpu-new/mednafen-gcc.h
index 9226242..b9ccba3 100644
--- a/src/hw_cpu/v810/fpu-new/mednafen-gcc.h
+++ b/src/hw_cpu/v810/fpu-new/mednafen-gcc.h
@@ -74,5 +74,5 @@ typedef int64_t sbits64;
 | a compiler does not support explicit inlining, this macro should be defined
 | to be `static'.
 *----------------------------------------------------------------------------*/
-#define INLINE extern inline
+#define INLINE static
 
