diff -Naur pytube-0.0.11.4.orig/pytube.py pytube-0.0.11.4.new/pytube.py
--- pytube-0.0.11.4.orig/pytube.py	2008-07-28 16:30:39.000000000 +0200
+++ pytube-0.0.11.4.new/pytube.py	2008-07-28 17:06:51.000000000 +0200
@@ -740,6 +740,7 @@
 		if not(os.path.isfile(self.userprefsfile)):
 			print "Configuration file not found, creating one now."
 			self.LoadDefaults()
+			self.WritePreferences()
 		else:
 			print "Configuration file found"
 			self.LoadPrefFile()
@@ -869,8 +870,8 @@
                          self.error.signal_autoconnect({"on_button1_clicked" : self.CloseError})
                          self.error.get_widget("errorlabel").set_text("No video was selected.")
 	def LoadPrefFile(self):
+		self.LoadDefaults()
 		self.configfile = open(self.userprefsfile, 'r')
-		self.configops = {}
 		x=-1
 		for i in self.configfile.readlines():
 			if x==-1:
@@ -886,33 +887,21 @@
 				x+=1
 		if x==-1:
 			self.configfile.close()
-			self.LoadDefaults()		
+			self.WritePreferences()
+			print "New configuration file created"
 		else:
 			self.configfile.close()
+			print "Loaded PyTube with preferences"
 		print "Configuration: " + str(self.configops)
 	def LoadDefaults(self):
-		self.configfile = open(self.userprefsfile, 'w')
 		self.configops = {}
-		self.configops[0] = "0"
-		self.configops[1] = "0"
-		self.configops[2] = str(os.path.expanduser("~"))
-		self.configops[3] = "0"
-		self.configops[4] = ""
-		self.configops[5] = ""
-		self.configops[6] = str("/usr/bin")
-		self.configfile.write(str("#valid configuration file for pytube 0.0.10.6\n"))
-		self.configfile.write(str(self.configops[0]) + "\n")
-		self.configfile.write(str(self.configops[1]) + "\n")
-		self.configfile.write(str(self.configops[2]) + "\n")
-		self.configfile.write(str(self.configops[3]) + "\n")
-		self.configfile.write(str(self.configops[4]) + "\n")
-		self.configfile.write(str(self.configops[5]) + "\n")
-		self.configfile.write(str(self.configops[6]) + "\n")
-		self.configfile.write("\n")
-		self.configfile.write("\n")
-		self.configfile.close()
-		os.chmod(self.userprefsfile,0600)
-		self.LoadPrefFile()
+		self.configops[0] = "0"	# Keep the temp. video after download
+		self.configops[1] = "0"	# Generate Thumbnail
+		self.configops[2] = str(os.path.expanduser("~")) # Save files in
+		self.configops[3] = "0"	# Use my youtube account
+		self.configops[4] = ""	# Youtube account user
+		self.configops[5] = ""	# Youtube account password
+		self.configops[6] = str("/usr/bin")	# Command directory.
 		print "Loaded PyTube with defaults values"
 		
 	def SavePrefFile(self):
@@ -924,30 +913,23 @@
 			self.configops[4] = str(self.preferences.get_widget("entry5").get_text())
 			self.configops[5] = str(self.preferences.get_widget("entry7").get_text())
 			self.configops[6] = str(self.preferences.get_widget("filechooserbutton7").get_filename())
-			self.configfile = open(self.userprefsfile, 'w')
-			self.configfile.write(str("#valid configuration file for pytube 0.0.10.6\n"))
-			self.configfile.write(str(self.configops[0]) + "\n")
-			self.configfile.write(str(self.configops[1]) + "\n")
-			self.configfile.write(str(self.configops[2]) + "\n")
-			self.configfile.write(str(self.configops[3]) + "\n")
-			self.configfile.write(str(self.configops[4]) + "\n")
-			self.configfile.write(str(self.configops[5]) + "\n")
-			self.configfile.write(str(self.configops[6]) + "\n")
-			self.configfile.close()
-			os.chmod(self.userprefsfile,0600)
+			self.WritePreferences()
 		elif self.configurationtype == "dependencies":
 			self.configops[6] = str(self.checkfiles.get_widget("filechooserbutton14").get_filename())
-                        self.configfile = open(self.userprefsfile, 'w')
-                        self.configfile.write(str("#valid configuration file for pytube 0.0.10.6\n"))
-                        self.configfile.write(str(self.configops[0]) + "\n")
-                        self.configfile.write(str(self.configops[1]) + "\n")
-                        self.configfile.write(str(self.configops[2]) + "\n")
-                        self.configfile.write(str(self.configops[3]) + "\n")
-                        self.configfile.write(str(self.configops[4]) + "\n")
-                        self.configfile.write(str(self.configops[5]) + "\n")
-                        self.configfile.write(str(self.configops[6]) + "\n")
-                        self.configfile.close()
-                        os.chmod(self.userprefsfile,0600)
+			self.WritePreferences()
+
+	def WritePreferences(self):
+		self.configfile = open(self.userprefsfile, 'w')
+		self.configfile.write(str("#valid configuration file for pytube 0.0.10.6\n"))
+		self.configfile.write(str(self.configops[0]) + "\n")
+		self.configfile.write(str(self.configops[1]) + "\n")
+		self.configfile.write(str(self.configops[2]) + "\n")
+		self.configfile.write(str(self.configops[3]) + "\n")
+		self.configfile.write(str(self.configops[4]) + "\n")
+		self.configfile.write(str(self.configops[5]) + "\n")
+		self.configfile.write(str(self.configops[6]) + "\n")
+		self.configfile.close()
+		os.chmod(self.userprefsfile,0600)
 
 	def ClosePreferences(self, widget):
 		self.configurationtype = "preferences"
