--- /var/tmp/portage/asciijump-1.0.2_beta/work/asciijump-1.0.2beta/Makefile.in	2003-12-14 05:54:47.000000000 -0600
+++ /var/tmp/portage/asciijump-1.0.2_beta/work/asciijump-1.0.2beta/Makefile-ctags	2004-06-23 14:32:58.808437408 -0500
@@ -101,7 +101,7 @@
 
 all: bin ${ac_name} ${as_name} ${other_obj}
 	@echo
-	@ctags ${other_src} ${as_src} ${ac_src} include/*
+	@exuberant-ctags ${other_src} ${as_src} ${ac_src} include/*
 	@echo "type: ${color}./${ac_name}${color_reset} to run asciijump client"
 	@echo "type: ${color}./${as_name}${color_reset} to run asciijump server"
 	
