diff --git a/src/trace.c b/src/trace.c
index 6477cef..a9f594b 100644
--- a/src/trace.c
+++ b/src/trace.c
@@ -167,7 +167,7 @@ trace (int daemonise,
 	}
 	if (set_value (dfd, "buffer_size_kb", 128000, NULL) < 0)
 		goto error;
-	if (set_value (dfd, "tracing_enabled",
+	if (set_value (dfd, "tracing_on",
 		       TRUE, &old_tracing_enabled) < 0)
 		goto error;
 
@@ -204,7 +204,7 @@ trace (int daemonise,
 	sigaction (SIGINT, &old_sigint, NULL);
 
 	/* Restore previous tracing settings */
-	if (set_value (dfd, "tracing_enabled",
+	if (set_value (dfd, "tracing_on",
 		       old_tracing_enabled, NULL) < 0)
 		goto error;
 	if (old_uselib_enabled >= 0)
