diff --git a/mime_tex.cpp b/mime_tex.cpp
index 4cdcb36..30e5e94 100644
--- a/mime_tex.cpp
+++ b/mime_tex.cpp
@@ -58,7 +58,7 @@ MimeTeX::MimeTeX::MimeTeX(QObject *parent)
 			"TeXformulaAction",
 			this,
 			SLOT(TeXActionActivated(QAction *, bool)),
-			KaduIcon(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/tex_icon.png")),
+			KaduIcon(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/tex_icon.png")),
 			tr("Insert TeX formula"));
 	
 	kdebugf2();
diff --git a/mime_tex_plugin.cpp b/mime_tex_plugin.cpp
index a766de0..ba4bcdd 100644
--- a/mime_tex_plugin.cpp
+++ b/mime_tex_plugin.cpp
@@ -22,7 +22,7 @@ int MimeTeX::MimeTeXPlugin::init(bool firstLoad)
 {
 	Q_UNUSED(firstLoad)
 
-	MainConfigurationWindow::registerUiFile(dataPath("kadu/plugins/configuration/mime_tex.ui"));
+	MainConfigurationWindow::registerUiFile(KaduPaths::instance()->pluginsLibPath() + QLatin1String("configuration/mime_tex.ui"));
 	MimeTeX::createInstance();
 
 	return 0;
@@ -31,7 +31,7 @@ int MimeTeX::MimeTeXPlugin::init(bool firstLoad)
 void MimeTeX::MimeTeXPlugin::done()
 {
 	MimeTeX::destroyInstance();
-	MainConfigurationWindow::unregisterUiFile(dataPath("kadu/plugins/configuration/mime_tex.ui"));
+	MainConfigurationWindow::unregisterUiFile(KaduPaths::instance()->pluginsLibPath() + QLatin1String("configuration/mime_tex.ui"));
 }
 
 Q_EXPORT_PLUGIN2(mime_tex, MimeTeX::MimeTeXPlugin)
diff --git a/mimetex/CMakeLists.txt b/mimetex/CMakeLists.txt
index 7405de6..24fad2c 100644
--- a/mimetex/CMakeLists.txt
+++ b/mimetex/CMakeLists.txt
@@ -10,4 +10,4 @@ STRING(REGEX REPLACE "-Werror(=.*\ )?" " " CMAKE_C_FLAGS ${CMAKE_C_FLAGS})
 
 add_executable(mimetex ${SOURCES})
 target_link_libraries(mimetex m)
-install (TARGETS mimetex DESTINATION ${KADU_LIBDIR}/kadu/plugins/bin/mime_tex)
+install (TARGETS mimetex DESTINATION ${KADU_PLUGINS_LIBDIR}/bin/mime_tex)
diff --git a/tex_formula_dialog.cpp b/tex_formula_dialog.cpp
index f7f55d0..9953458 100644
--- a/tex_formula_dialog.cpp
+++ b/tex_formula_dialog.cpp
@@ -43,14 +43,14 @@ MimeTeX::TeXFormulaDialog::TeXFormulaDialog(ChatWidget *parent, Qt::WindowFlags
 	
 	setAttribute(Qt::WA_DeleteOnClose);
 	// Check if .kadu/tex directory exists. Create one if not.
-	QDir TeXDir(profilePath("tex"));
+	QDir TeXDir(KaduPaths::instance()->profilePath() + QLatin1String("tex"));
 	if(!TeXDir.exists())
-		if(TeXDir.mkdir(profilePath("tex"))) {
-			kdebugm(KDEBUG_INFO, "Error creating directory %s\n", profilePath("tex").toLatin1().constData());
+		if(TeXDir.mkdir(KaduPaths::instance()->profilePath() + QLatin1String("tex"))) {
+			kdebugm(KDEBUG_INFO, "Error creating directory %s\n", KaduPaths::instance()->profilePath() + QLatin1String("tex").toLatin1().constData());
                 }
 	
 	QDateTime curTime = QDateTime::currentDateTime();
-	tmpFileName = profilePath("tex/formula_%1.gif").arg(curTime.toString("yyyy-MM-dd_hh-mm"));
+	tmpFileName = KaduPaths::instance()->profilePath() + QString("tex/formula_%1.gif").arg(curTime.toString("yyyy-MM-dd_hh-mm"));
 	if(QFile(tmpFileName).exists())
 	{
 		kdebugm(KDEBUG_INFO, "File %s already exists!!\n", tmpFileName.toLatin1().constData());
@@ -58,7 +58,7 @@ MimeTeX::TeXFormulaDialog::TeXFormulaDialog(ChatWidget *parent, Qt::WindowFlags
 		QString tmpStr;
 		do
 		{
-			tmpStr = profilePath("tex/formula_%1_%2.gif").arg(curTime.toString("yyyy-MM-dd_hh:mm")).arg(i++);
+			tmpStr = KaduPaths::instance()->profilePath() + QString("tex/formula_%1_%2.gif").arg(curTime.toString("yyyy-MM-dd_hh:mm")).arg(i++);
 			kdebugm(KDEBUG_INFO, "tmpStr: %s\n", tmpStr.toLatin1().constData());
 		}
 		while(QFile(tmpStr).exists());
@@ -101,19 +101,19 @@ MimeTeX::TeXFormulaDialog::TeXFormulaDialog(ChatWidget *parent, Qt::WindowFlags
 	symbolsView->setViewMode(QListWidget::IconMode);
 	
 	componentsView->addTab(relationsView, 
-				   QPixmap(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_icon.png")),
+				   QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_icon.png")),
 			       tr("&Relations"));
 	componentsView->addTab(greekLettersView,
-				   QPixmap(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_icon.png")),
+				   QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_icon.png")),
 			       tr("&Greek Letters"));
 	componentsView->addTab(arrowsView,
-				   QPixmap(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_icon.png")),
+				   QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_icon.png")),
 			       tr("&Arrows"));
 	componentsView->addTab(delimitersView,
-				   QPixmap(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_icon.png")),
+				   QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_icon.png")),
 			       tr("&Delimiters"));
 	componentsView->addTab(symbolsView,
-				   QPixmap(dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_icon.png")),
+				   QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_icon.png")),
 			       tr("&Symbols"));
 	
 	connect(relationsView, SIGNAL(itemClicked(QListWidgetItem *)), this, SLOT(insertComponentSlot(QListWidgetItem *)));
@@ -124,429 +124,429 @@ MimeTeX::TeXFormulaDialog::TeXFormulaDialog(ChatWidget *parent, Qt::WindowFlags
 	
 	// Filling Relations tab
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_frac.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_frac.png"))),
 		"\\frac{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_index_upper.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_index_upper.png"))),
 		"^{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_index_lower.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_index_lower.png"))),
 		"_{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sqrt.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sqrt.png"))),
 		"\\sqrt{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sqrt_n.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sqrt_n.png"))),
 		"\\sqrt[]{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_geq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_geq.png"))),
 		"\\geq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_leq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_leq.png"))),
 		"\\leq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sim.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sim.png"))),
 		"\\sim", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_neq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_neq.png"))),
 		"\\neq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_equiv.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_equiv.png"))),
 		"\\equiv", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_approx.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_approx.png"))),
 		"\\approx", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_simeq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_simeq.png"))),
 		"\\simeq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_ll.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_ll.png"))),
 		"\\ll", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_gg.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_gg.png"))),
 		"\\gg", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_pm.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_pm.png"))),
 		"\\pm", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_in.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_in.png"))),
 		"\\in", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_ni.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_ni.png"))),
 		"\\ni", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_cap.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_cap.png"))),
 		"\\cap", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_cup.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_cup.png"))),
 		"\\cup", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_bigcap.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_bigcap.png"))),
 		"\\bigcap", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_bigcup.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_bigcup.png"))),
 		"\\bigcup", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_subset.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_subset.png"))),
 		"\\subset", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_supset.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_supset.png"))),
 		"\\supset", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_subseteq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_subseteq.png"))),
 		"\\subseteq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_supseteq.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_supseteq.png"))),
 		"\\supseteq", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sum.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sum.png"))),
 		"\\sum{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sum_from.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sum_from.png"))),
 		"\\sum\\limits_{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_sum_from_to.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_sum_from_to.png"))),
 		"\\sum\\limits_{}^{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_prod.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_prod.png"))),
 		"\\prod{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_prod_from.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_prod_from.png"))),
 		"\\prod\\limits_{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_prod_from_to.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_prod_from_to.png"))),
 		"\\prod\\limits_{}^{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_int.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_int.png"))),
 		"\\int{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_int_from.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_int_from.png"))),
 		"\\int\\limits_{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_int_from_to.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_int_from_to.png"))),
 		"\\int\\limits_{}^{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_oint.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_oint.png"))),
 		"\\oint{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_oint_from.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_oint_from.png"))),
 		"\\oint\\limits_{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_oint_from_to.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_oint_from_to.png"))),
 		"\\oint\\limits_{}^{}{}", relationsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/relations_tab_lim.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/relations_tab_lim.png"))),
 		"\\lim_{}{}", relationsView);
 	
 	// Filling Greek Letters tab
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_alpha.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_alpha.png"))),
 		"\\alpha", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_beta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_beta.png"))),
 		"\\beta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_gamma.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_gamma.png"))),
 		"\\gamma", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Gamma.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Gamma.png"))),
 		"\\Gamma", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_delta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_delta.png"))),
 		"\\delta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Delta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Delta.png"))),
 		"\\Delta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_epsilon.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_epsilon.png"))),
 		"\\epsilon", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_varepsilon.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_varepsilon.png"))),
 		"\\varepsilon", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_zeta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_zeta.png"))),
 		"\\zeta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_eta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_eta.png"))),
 		"\\eta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_theta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_theta.png"))),
 		"\\theta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_vartheta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_vartheta.png"))),
 		"\\vartheta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Theta.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Theta.png"))),
 		"\\Theta", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_iota.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_iota.png"))),
 		"\\iota", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_kappa.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_kappa.png"))),
 		"\\kappa", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_lambda.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_lambda.png"))),
 		"\\lambda", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Lambda.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Lambda.png"))),
 			"\\Lambda", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_mu.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_mu.png"))),
 		"\\mu", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_nu.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_nu.png"))),
 		"\\nu", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_xi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_xi.png"))),
 		"\\xi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Xi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Xi.png"))),
 		"\\Xi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_pi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_pi.png"))),
 		"\\pi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_varpi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_varpi.png"))),
 		"\\varpi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Pi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Pi.png"))),
 		"\\Pi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_rho.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_rho.png"))),
 		"\\rho", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_varrho.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_varrho.png"))),
 		"\\varrho", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_sigma.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_sigma.png"))),
 		"\\sigma", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_varsigma.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_varsigma.png"))),
 		"\\varsigma", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Sigma.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Sigma.png"))),
 		"\\Sigma", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_tau.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_tau.png"))),
 		"\\tau", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_upsilon.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_upsilon.png"))),
 		"\\upsilon", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_phi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_phi.png"))),
 		"\\phi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_varphi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_varphi.png"))),
 		"\\varphi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Phi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Phi.png"))),
 		"\\Phi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_chi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_chi.png"))),
 		"\\chi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_psi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_psi.png"))),
 		"\\psi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Psi.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Psi.png"))),
 		"\\Psi", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_omega.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_omega.png"))),
 		"\\omega", greekLettersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/greek_letters_tab_Omega.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/greek_letters_tab_Omega.png"))),
 		"\\Omega", greekLettersView);
 
 	// Filling arrows tab
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_vec.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_vec.png"))),
 		"\\vec{}", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_swarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_swarrow.png"))),
 		"\\swarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_leftarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_leftarrow.png"))),
 		"\\leftarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_nwarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_nwarrow.png"))),
 		"\\nwarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_uparrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_uparrow.png"))),
 		"\\uparrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_nearrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_nearrow.png"))),
 		"\\nearrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_rightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_rightarrow.png"))),
 		"\\rightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_searrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_searrow.png"))),
 		"\\searrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_leftrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_leftrightarrow.png"))),
 		"\\leftrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_updownarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_updownarrow.png"))),
 		"\\updownarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_longleftarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_longleftarrow.png"))),
 		"\\longleftarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_longrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_longrightarrow.png"))),
 		"\\longrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_longleftrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_longleftrightarrow.png"))),
 		"\\longleftrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Leftarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Leftarrow.png"))),
 		"\\Leftarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Rightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Rightarrow.png"))),
 		"\\Rightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Uparrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Uparrow.png"))),
 		"\\Uparrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Downarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Downarrow.png"))),
 		"\\Downarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Leftrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Leftrightarrow.png"))),
 		"\\Leftrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Longleftarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Longleftarrow.png"))),
 		"\\Longleftarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Longrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Longrightarrow.png"))),
 		"\\Longrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_Longleftrightarrow.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_Longleftrightarrow.png"))),
 		"\\Longleftrightarrow", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_leftharpoonup.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_leftharpoonup.png"))),
 		"\\leftharpoonup", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_leftharpoondown.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_leftharpoondown.png"))),
 		"\\leftharpoondown", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_rightharpoonup.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_rightharpoonup.png"))),
 		"\\rightharpoonup", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_rightharpoondown.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_rightharpoondown.png"))),
 		"\\rightharpoondown", arrowsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/arrows_tab_rightleftharpoons.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/arrows_tab_rightleftharpoons.png"))),
 		"\\rightleftharpoons", arrowsView);
 
 	// Filling Delimiters tab
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_lbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_lbrace.png"))),
 		"\\lbrace", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_rbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_rbrace.png"))),
 		"\\rbrace", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_langle.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_langle.png"))),
 		"\\langle", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_rangle.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_rangle.png"))),
 		"\\rangle", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_hat.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_hat.png"))),
 		"\\hat", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_overbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_overbrace.png"))),
 		"\\overbrace{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_overbrace_desc.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_overbrace_desc.png"))),
 		"\\overbrace{}^{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_underbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_underbrace.png"))),
 		"\\underbrace{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_underbrace_desc.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_underbrace_desc.png"))),
 		"\\underbrace{}_{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_overline.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_overline.png"))),
 		"\\overline{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_underline.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_underline.png"))),
 		"\\underline{}", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_left_bracket.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_left_bracket.png"))),
 		"\\left(", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_right_bracket.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_right_bracket.png"))),
 		"\\right)", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_left_square_bracket.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_left_square_bracket.png"))),
 		"\\left[", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_right_square_bracket.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_right_square_bracket.png"))),
 		"\\right]", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_left_lbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_left_lbrace.png"))),
 		"\\left{", delimitersView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/delimiters_tab_right_rbrace.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/delimiters_tab_right_rbrace.png"))),
 		"\\right}", delimitersView);
 
 	// Filling Symbols tab
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_infty.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_infty.png"))),
 		"\\infty", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_cdot.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_cdot.png"))),
 		"\\cdot", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_cdots.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_cdots.png"))),
 		"\\cdots", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_ldots.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_ldots.png"))),
 		"\\ldots", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_ddots.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_ddots.png"))),
 		"\\ddots", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_hbar.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_hbar.png"))),
 		"\\hbar", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_nabla.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_nabla.png"))),
 		"\\nabla", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_partial.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_partial.png"))),
 		"\\partial", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_aleph.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_aleph.png"))),
 		"\\aleph", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_Re.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_Re.png"))),
 		"\\Re", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_Im.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_Im.png"))),
 		"\\Im", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_bot.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_bot.png"))),
 		"\\bot", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_emptyset.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_emptyset.png"))),
 		"\\emptyset", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_prime.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_prime.png"))),
 		"\\prime", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_surd.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_surd.png"))),
 		"\\surd", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_angle.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_angle.png"))),
 		"\\angle", symbolsView);
 	(void) new QListWidgetItem(QIcon(QPixmap(
-		dataPath("kadu/plugins/data/mime_tex/mime_tex_icons/symbols_tab_forall.png"))),
+		KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/mime_tex_icons/symbols_tab_forall.png"))),
 		"\\forall", symbolsView);
 
 	
-	undoButton = new QPushButton(QIcon(QPixmap(dataPath("kadu/plugins/data/mime_tex/editor_icons/undo.png"))),
+	undoButton = new QPushButton(QIcon(QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/editor_icons/undo.png"))),
 			tr("&Undo"), this);
-	redoButton = new QPushButton(QIcon(QPixmap(dataPath("kadu/plugins/data/mime_tex/editor_icons/redo.png"))),
+	redoButton = new QPushButton(QIcon(QPixmap(KaduPaths::instance()->pluginsLibPath() + QLatin1String("data/mime_tex/editor_icons/redo.png"))),
 			tr("&Redo"), this);
 	QPushButton *okButton = new QPushButton(tr("&Ok"), this);
 	QPushButton *cancelButton = new QPushButton(tr("&Cancel"), this);
@@ -625,7 +625,7 @@ void MimeTeX::TeXFormulaDialog::timeoutSlot()
 		for(it = arguments.constBegin(); it != arguments.constEnd(); ++it)
 			kdebugm(KDEBUG_INFO, "Arg[%d]=%s\n", i++, (*it).toLocal8Bit().constData());
 #endif
-		mimeTeXProcess.start(libPath("kadu/plugins/bin/mime_tex/mimetex"),
+		mimeTeXProcess.start(KaduPaths::instance()->pluginsLibPath() + QLatin1String("bin/mime_tex/mimetex"),
 			arguments);
 		mimeTeXProcess.waitForStarted();
 	}
