diff -ruN mime_tex/mime_tex.cpp mime_tex.new/mime_tex.cpp
--- mime_tex/mime_tex.cpp	2008-02-01 16:38:55.000000000 +0100
+++ mime_tex.new/mime_tex.cpp	2008-02-20 23:03:51.000000000 +0100
@@ -13,7 +13,7 @@
 
 #include "mime_tex.h"
 #include "tex_formula_dialog.h"
-#include "ui_handler.h"
+#include "mime_tex_ui_handler.h"
 
 #include "icons_manager.h"
 #include "misc.h"
diff -ruN mime_tex/mime_tex_ui_handler.cpp mime_tex.new/mime_tex_ui_handler.cpp
--- mime_tex/mime_tex_ui_handler.cpp	1970-01-01 01:00:00.000000000 +0100
+++ mime_tex.new/mime_tex_ui_handler.cpp	2008-02-20 23:03:16.000000000 +0100
@@ -0,0 +1,29 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "configuration_window_widgets.h"
+#include "debug.h"
+
+#include "mime_tex_ui_handler.h"
+
+MimeTeX::UIHandler::UIHandler()
+{
+	kdebugf();
+}
+
+MimeTeX::UIHandler::~UIHandler()
+{
+	kdebugf();
+}
+
+void MimeTeX::UIHandler::mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow)
+{
+	kdebugf();
+}
+
diff -ruN mime_tex/mime_tex_ui_handler.h mime_tex.new/mime_tex_ui_handler.h
--- mime_tex/mime_tex_ui_handler.h	1970-01-01 01:00:00.000000000 +0100
+++ mime_tex.new/mime_tex_ui_handler.h	2008-02-20 23:03:06.000000000 +0100
@@ -0,0 +1,36 @@
+#ifndef MIME_TEX_UI_HANDLER_H
+#define MIME_TEX_UI_HANDLER_H
+
+#include "main_configuration_window.h"
+
+namespace MimeTeX {
+	/*!
+	 * \brief Handler of configuration dialogs GUI
+	 */
+	class UIHandler : public ConfigurationUiHandler
+	{
+		Q_OBJECT
+
+		public:
+			//! Default constructor
+			UIHandler();
+
+			//! Default destructor
+			~UIHandler();
+
+			public slots:
+				/*!
+				 * This slot is called when the configuration
+				 * dialog is created. It handles setting
+				 * appropriate value of the font size combo
+				 * box
+				 *
+				 * \param mainConfigurationWindow Pointer to
+				 *        the configuration window object
+				 */
+				virtual void mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow);
+
+	};
+}
+
+#endif /* MIME_TEX_UI_HANDLER_H */
diff -ruN mime_tex/spec mime_tex.new/spec
--- mime_tex/spec	2008-02-01 16:38:55.000000000 +0100
+++ mime_tex.new/spec	2008-02-20 23:03:32.000000000 +0100
@@ -1,4 +1,4 @@
-MODULE_SOURCES="mime_tex.cpp tex_formula_dialog.cpp formula_view.cpp formula_widget.cpp ui_handler.cpp"
-MODULE_MOC_HEADERS="mime_tex.h tex_formula_dialog.h formula_view.h formula_widget.h ui_handler.h"
+MODULE_SOURCES="mime_tex.cpp tex_formula_dialog.cpp formula_view.cpp formula_widget.cpp mime_tex_ui_handler.cpp"
+MODULE_MOC_HEADERS="mime_tex.h tex_formula_dialog.h formula_view.h formula_widget.h mime_tex_ui_handler.h"
 MODULE_3RDPARTY=mimetex
 
diff -ruN mime_tex/ui_handler.cpp mime_tex.new/ui_handler.cpp
--- mime_tex/ui_handler.cpp	2008-02-01 16:38:55.000000000 +0100
+++ mime_tex.new/ui_handler.cpp	1970-01-01 01:00:00.000000000 +0100
@@ -1,29 +0,0 @@
-/***************************************************************************
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- *                                                                         *
- ***************************************************************************/
-
-#include "configuration_window_widgets.h"
-#include "debug.h"
-
-#include "ui_handler.h"
-
-MimeTeX::UIHandler::UIHandler()
-{
-	kdebugf();
-}
-
-MimeTeX::UIHandler::~UIHandler()
-{
-	kdebugf();
-}
-
-void MimeTeX::UIHandler::mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow)
-{
-	kdebugf();
-}
-
diff -ruN mime_tex/ui_handler.h mime_tex.new/ui_handler.h
--- mime_tex/ui_handler.h	2008-02-01 16:38:55.000000000 +0100
+++ mime_tex.new/ui_handler.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,36 +0,0 @@
-#ifndef UI_HANDLER_H
-#define UI_HANDLER_H
-
-#include "main_configuration_window.h"
-
-namespace MimeTeX {
-	/*!
-	 * \brief Handler of configuration dialogs GUI
-	 */
-	class UIHandler : public ConfigurationUiHandler
-	{
-		Q_OBJECT
-
-		public:
-			//! Default constructor
-			UIHandler();
-
-			//! Default destructor
-			~UIHandler();
-
-			public slots:
-				/*!
-				 * This slot is called when the configuration
-				 * dialog is created. It handles setting
-				 * appropriate value of the font size combo
-				 * box
-				 *
-				 * \param mainConfigurationWindow Pointer to
-				 *        the configuration window object
-				 */
-				virtual void mainConfigurationWindowCreated(MainConfigurationWindow *mainConfigurationWindow);
-
-	};
-}
-
-#endif /* UI_HANDLER_H */
